// @revenuecat/purchases-ui-js/dist/components/text/text-utils.js
import { getSizeStyle, getTextComponentTag, getTextStyles, } from "../../utils";
import { getComponentStyles } from "../../utils";
/**
 * Generates comprehensive styles for text components by combining text, component and size styles
 * @param props - Text component properties including font, color, background, spacing etc.
 * @returns Object containing text style variables and the appropriate HTML tag to render
 */
export const getTextComponentStyles = (props) => {
    const { font_size = "body_m", color = {
        dark: { type: "hex", value: "#000000" },
        light: { type: "hex", value: "#000000" },
    }, background_color = {
        dark: { type: "hex", value: "#FFFFFF" },
        light: { type: "hex", value: "#FFFFFF" },
    }, padding, margin, colorMode, size, } = props;
    const tagToRender = getTextComponentTag(font_size);
    const textStyles = getTextStyles(props);
    const componentStyles = getComponentStyles({
        backgroundColor: background_color,
        textColor: color,
        margin,
        padding,
        colorMode,
    });
    const sizeStyles = getSizeStyle(size);
    return {
        textStyles: {
            ...textStyles,
            ...componentStyles,
            ...sizeStyles,
        },
        tagToRender,
    };
};
