// @revenuecat/purchases-ui-js/dist/types.js
// Text types
export var FontSizeTags;
(function (FontSizeTags) {
    FontSizeTags["body_s"] = "p";
    FontSizeTags["body_m"] = "p";
    FontSizeTags["body_l"] = "p";
    FontSizeTags["body_xl"] = "p";
    FontSizeTags["heading_xs"] = "h6";
    FontSizeTags["heading_s"] = "h5";
    FontSizeTags["heading_m"] = "h4";
    FontSizeTags["heading_l"] = "h3";
    FontSizeTags["heading_xl"] = "h2";
    FontSizeTags["heading_xxl"] = "h1";
})(FontSizeTags || (FontSizeTags = {}));
export var FontWeights;
(function (FontWeights) {
    FontWeights[FontWeights["extra_light"] = 100] = "extra_light";
    FontWeights[FontWeights["thin"] = 200] = "thin";
    FontWeights[FontWeights["light"] = 300] = "light";
    FontWeights[FontWeights["regular"] = 400] = "regular";
    FontWeights[FontWeights["medium"] = 500] = "medium";
    FontWeights[FontWeights["semi_bold"] = 600] = "semi_bold";
    FontWeights[FontWeights["bold"] = 700] = "bold";
    FontWeights[FontWeights["extra_bold"] = 800] = "extra_bold";
    FontWeights[FontWeights["black"] = 900] = "black";
})(FontWeights || (FontWeights = {}));
export var FontSizes;
(function (FontSizes) {
    FontSizes["body_s"] = "13px";
    FontSizes["body_m"] = "14px";
    FontSizes["body_l"] = "15px";
    FontSizes["body_xl"] = "16px";
    FontSizes["heading_xs"] = "14px";
    FontSizes["heading_s"] = "16px";
    FontSizes["heading_m"] = "18px";
    FontSizes["heading_l"] = "20px";
    FontSizes["heading_xl"] = "26px";
    FontSizes["heading_xxl"] = "32px";
})(FontSizes || (FontSizes = {}));
export var StackDirection;
(function (StackDirection) {
    StackDirection["vertical"] = "column";
    StackDirection["horizontal"] = "row";
})(StackDirection || (StackDirection = {}));
export var StackAlignment;
(function (StackAlignment) {
    StackAlignment["top"] = "flex-start";
    StackAlignment["leading"] = "flex-start";
    StackAlignment["center"] = "center";
    StackAlignment["bottom"] = "flex-end";
    StackAlignment["trailing"] = "flex-end";
})(StackAlignment || (StackAlignment = {}));
export var StackDistribution;
(function (StackDistribution) {
    StackDistribution["space_between"] = "space-between";
    StackDistribution["space_around"] = "space-around";
    StackDistribution["space_evenly"] = "space-evenly";
    StackDistribution["start"] = "flex-start";
    StackDistribution["center"] = "center";
    StackDistribution["end"] = "flex-end";
})(StackDistribution || (StackDistribution = {}));
