// @revenuecat/purchases-js/dist/Purchases.es.js
var Xi = Object.create;
var un = Object.defineProperty;
var Ji = Object.getOwnPropertyDescriptor;
var ar = (e, t) => (t = Symbol[e]) ? t : Symbol.for("Symbol." + e), pt = (e) => {
  throw TypeError(e);
};
var sr = (e, t, n) => t in e ? un(e, t, { enumerable: !0, configurable: !0, writable: !0, value: n }) : e[t] = n;
var rr = (e, t) => un(e, "name", { value: t, configurable: !0 });
var lr = (e) => [, , , Xi((e == null ? void 0 : e[ar("metadata")]) ?? null)], cr = ["class", "method", "getter", "setter", "accessor", "field", "value", "get", "set"], ht = (e) => e !== void 0 && typeof e != "function" ? pt("Function expected") : e, eo = (e, t, n, r, i) => ({ kind: cr[e], name: t, metadata: r, addInitializer: (o) => n._ ? pt("Already initialized") : i.push(ht(o || null)) }), fn = (e, t) => sr(t, ar("metadata"), e[3]), ur = (e, t, n, r) => {
  for (var i = 0, o = e[t >> 1], a = o && o.length; i < a; i++) t & 1 ? o[i].call(n) : r = o[i].call(n, r);
  return r;
}, fr = (e, t, n, r, i, o) => {
  var a, s, c, l, d, u = t & 7, f = !!(t & 8), v = !!(t & 16), p = u > 3 ? e.length + 1 : u ? f ? 1 : 2 : 0, b = cr[u + 5], x = u > 3 && (e[p - 1] = []), m = e[p] || (e[p] = []), E = u && (!v && !f && (i = i.prototype), u < 5 && (u > 3 || !v) && Ji(u < 4 ? i : { get [n]() {
    return ir(this, o);
  }, set [n](h) {
    return or(this, o, h);
  } }, n));
  u ? v && u < 4 && rr(o, (u > 2 ? "set " : u > 1 ? "get " : "") + n) : rr(i, n);
  for (var P = r.length - 1; P >= 0; P--)
    l = eo(u, n, c = {}, e[3], m), u && (l.static = f, l.private = v, d = l.access = { has: v ? (h) => to(i, h) : (h) => n in h }, u ^ 3 && (d.get = v ? (h) => (u ^ 1 ? ir : no)(h, i, u ^ 4 ? o : E.get) : (h) => h[n]), u > 2 && (d.set = v ? (h, w) => or(h, i, w, u ^ 4 ? o : E.set) : (h, w) => h[n] = w)), s = (0, r[P])(u ? u < 4 ? v ? o : E[b] : u > 4 ? void 0 : { get: E.get, set: E.set } : i, l), c._ = 1, u ^ 4 || s === void 0 ? ht(s) && (u > 4 ? x.unshift(s) : u ? v ? o = s : E[b] = s : i = s) : typeof s != "object" || s === null ? pt("Object expected") : (ht(a = s.get) && (E.get = a), ht(a = s.set) && (E.set = a), ht(a = s.init) && x.unshift(a));
  return u || fn(e, i), E && un(i, n, E), v ? u ^ 4 ? o : E : i;
}, R = (e, t, n) => sr(e, typeof t != "symbol" ? t + "" : t, n), dn = (e, t, n) => t.has(e) || pt("Cannot " + n), to = (e, t) => Object(t) !== t ? pt('Cannot use the "in" operator on this value') : e.has(t), ir = (e, t, n) => (dn(e, t, "read from private field"), n ? n.call(e) : t.get(e));
var or = (e, t, n, r) => (dn(e, t, "write to private field"), r ? r.call(e, n) : t.set(e, n), n), no = (e, t, n) => (dn(e, t, "access private method"), n);
const ro = "5";
typeof window < "u" && (window.__svelte || (window.__svelte = { v: /* @__PURE__ */ new Set() })).v.add(ro);
let ot = !1;
function io() {
  ot = !0;
}
io();
const Ln = 1, Un = 2, qr = 4, oo = 8, ao = 16, so = 1, lo = 2, co = 4, uo = 8, fo = 16, vo = 1, go = 2, fe = Symbol(), Dn = "http://www.w3.org/2000/svg", ho = /[&"<]/g;
function po(e, t) {
  const n = String(e ?? ""), r = ho;
  r.lastIndex = 0;
  let i = "", o = 0;
  for (; r.test(n); ) {
    const a = r.lastIndex - 1, s = n[a];
    i += n.substring(o, a) + (s === "&" ? "&amp;" : s === '"' ? "&quot;" : "&lt;"), o = a + 1;
  }
  return i + n.substring(o);
}
const _o = !1;
var Ot = Array.isArray, Mn = Array.from, bo = Object.defineProperty, Se = Object.getOwnPropertyDescriptor, jr = Object.getOwnPropertyDescriptors, mo = Object.prototype, yo = Array.prototype, $t = Object.getPrototypeOf;
function _t(e) {
  return typeof e == "function";
}
const Ce = () => {
};
function wo(e) {
  return e();
}
function hn(e) {
  for (var t = 0; t < e.length; t++)
    e[t]();
}
const Re = 2, Fr = 4, Nt = 8, Cn = 16, be = 32, St = 64, We = 128, Kt = 256, oe = 512, je = 1024, at = 2048, xe = 4096, Rt = 8192, Eo = 16384, st = 32768, xo = 65536, Io = 1 << 18, Hr = 1 << 19, Be = Symbol("$state"), Vr = Symbol("legacy props"), Po = Symbol("");
function Gr(e) {
  return e === this.v;
}
function zr(e, t) {
  return e != e ? t == t : e !== t || e !== null && typeof e == "object" || typeof e == "function";
}
function Bn(e) {
  return !zr(e, this.v);
}
function ko(e) {
  throw new Error("effect_in_teardown");
}
function To() {
  throw new Error("effect_in_unowned_derived");
}
function Ao(e) {
  throw new Error("effect_orphan");
}
function Oo() {
  throw new Error("effect_update_depth_exceeded");
}
function No(e) {
  throw new Error("lifecycle_legacy_only");
}
function So(e) {
  throw new Error("props_invalid_value");
}
function Ro() {
  throw new Error("state_descriptors_fixed");
}
function Lo() {
  throw new Error("state_prototype_fixed");
}
function Uo() {
  throw new Error("state_unsafe_local_read");
}
function Do() {
  throw new Error("state_unsafe_mutation");
}
function pe(e) {
  return {
    f: 0,
    // TODO ideally we could skip this altogether, but it causes type errors
    v: e,
    reactions: null,
    equals: Gr,
    version: 0
  };
}
// @__NO_SIDE_EFFECTS__
function Jt(e, t = !1) {
  var r;
  const n = pe(e);
  return t || (n.equals = Bn), ot && L !== null && L.l !== null && ((r = L.l).s ?? (r.s = [])).push(n), n;
}
function he(e, t = !1) {
  return /* @__PURE__ */ Mo(/* @__PURE__ */ Jt(e, t));
}
// @__NO_SIDE_EFFECTS__
function Mo(e) {
  return q !== null && q.f & Re && (Ie === null ? Ho([e]) : Ie.push(e)), e;
}
function O(e, t) {
  return q !== null && tn() && q.f & (Re | Cn) && // If the source was created locally within the current derived, then
  // we allow the mutation.
  (Ie === null || !Ie.includes(e)) && Do(), pn(e, t);
}
function pn(e, t) {
  return e.equals(t) || (e.v = t, e.version = ii(), Yr(e, je), tn() && S !== null && S.f & oe && !(S.f & be) && (ee !== null && ee.includes(e) ? (me(S, je), nn(S)) : qe === null ? Vo([e]) : qe.push(e))), t;
}
function Yr(e, t) {
  var n = e.reactions;
  if (n !== null)
    for (var r = tn(), i = n.length, o = 0; o < i; o++) {
      var a = n[o], s = a.f;
      s & je || !r && a === S || (me(a, t), s & (oe | We) && (s & Re ? Yr(
        /** @type {Derived} */
        a,
        at
      ) : nn(
        /** @type {Effect} */
        a
      )));
    }
}
// @__NO_SIDE_EFFECTS__
function K(e) {
  var t = Re | je;
  S === null ? t |= We : S.f |= Hr;
  const n = {
    children: null,
    ctx: L,
    deps: null,
    equals: Gr,
    f: t,
    fn: e,
    reactions: null,
    v: (
      /** @type {V} */
      null
    ),
    version: 0,
    parent: S
  };
  if (q !== null && q.f & Re) {
    var r = (
      /** @type {Derived} */
      q
    );
    (r.children ?? (r.children = [])).push(n);
  }
  return n;
}
// @__NO_SIDE_EFFECTS__
function ge(e) {
  const t = /* @__PURE__ */ K(e);
  return t.equals = Bn, t;
}
function $r(e) {
  var t = e.children;
  if (t !== null) {
    e.children = null;
    for (var n = 0; n < t.length; n += 1) {
      var r = t[n];
      r.f & Re ? qn(
        /** @type {Derived} */
        r
      ) : ke(
        /** @type {Effect} */
        r
      );
    }
  }
}
function Kr(e) {
  var t, n = S;
  Pe(e.parent);
  try {
    $r(e), t = oi(e);
  } finally {
    Pe(n);
  }
  return t;
}
function Qr(e) {
  var t = Kr(e), n = (nt || e.f & We) && e.deps !== null ? at : oe;
  me(e, n), e.equals(t) || (e.v = t, e.version = ii());
}
function qn(e) {
  $r(e), Tt(e, 0), me(e, Rt), e.v = e.children = e.deps = e.ctx = e.reactions = null;
}
function Wr(e) {
  S === null && q === null && Ao(), q !== null && q.f & We && To(), Gn && ko();
}
function Co(e, t) {
  var n = t.last;
  n === null ? t.last = t.first = e : (n.next = e, e.prev = n, t.last = e);
}
function lt(e, t, n, r = !0) {
  var i = (e & St) !== 0, o = S, a = {
    ctx: L,
    deps: null,
    deriveds: null,
    nodes_start: null,
    nodes_end: null,
    f: e | je,
    first: null,
    fn: t,
    last: null,
    next: null,
    parent: i ? null : o,
    prev: null,
    teardown: null,
    transitions: null,
    version: 0
  };
  if (n) {
    var s = rt;
    try {
      dr(!0), Ut(a), a.f |= Eo;
    } catch (d) {
      throw ke(a), d;
    } finally {
      dr(s);
    }
  } else t !== null && nn(a);
  var c = n && a.deps === null && a.first === null && a.nodes_start === null && a.teardown === null && (a.f & Hr) === 0;
  if (!c && !i && r && (o !== null && Co(a, o), q !== null && q.f & Re)) {
    var l = (
      /** @type {Derived} */
      q
    );
    (l.children ?? (l.children = [])).push(a);
  }
  return a;
}
function jn(e) {
  const t = lt(Nt, null, !1);
  return me(t, oe), t.teardown = e, t;
}
function _n(e) {
  Wr();
  var t = S !== null && (S.f & be) !== 0 && L !== null && !L.m;
  if (t) {
    var n = (
      /** @type {ComponentContext} */
      L
    );
    (n.e ?? (n.e = [])).push({
      fn: e,
      effect: S,
      reaction: q
    });
  } else {
    var r = Fn(e);
    return r;
  }
}
function Bo(e) {
  return Wr(), Lt(e);
}
function qo(e) {
  const t = lt(St, e, !0);
  return () => {
    ke(t);
  };
}
function Fn(e) {
  return lt(Fr, e, !1);
}
function Ke(e, t) {
  var n = (
    /** @type {ComponentContextLegacy} */
    L
  ), r = { effect: null, ran: !1 };
  n.l.r1.push(r), r.effect = Lt(() => {
    e(), !r.ran && (r.ran = !0, O(n.l.r2, !0), Qe(t));
  });
}
function Hn() {
  var e = (
    /** @type {ComponentContextLegacy} */
    L
  );
  Lt(() => {
    if (g(e.l.r2)) {
      for (var t of e.l.r1) {
        var n = t.effect;
        n.f & oe && me(n, at), ut(n) && Ut(n), t.ran = !1;
      }
      e.l.r2.v = !1;
    }
  });
}
function Lt(e) {
  return lt(Nt, e, !0);
}
function B(e) {
  return Ze(e);
}
function Ze(e, t = 0) {
  return lt(Nt | Cn | t, e, !0);
}
function Le(e, t = !0) {
  return lt(Nt | be, e, !0, t);
}
function Zr(e) {
  var t = e.teardown;
  if (t !== null) {
    const n = Gn, r = q;
    vr(!0), Fe(null);
    try {
      t.call(null);
    } finally {
      vr(n), Fe(r);
    }
  }
}
function Xr(e) {
  var t = e.deriveds;
  if (t !== null) {
    e.deriveds = null;
    for (var n = 0; n < t.length; n += 1)
      qn(t[n]);
  }
}
function Jr(e, t = !1) {
  var n = e.first;
  for (e.first = e.last = null; n !== null; ) {
    var r = n.next;
    ke(n, t), n = r;
  }
}
function jo(e) {
  for (var t = e.first; t !== null; ) {
    var n = t.next;
    t.f & be || ke(t), t = n;
  }
}
function ke(e, t = !0) {
  var n = !1;
  if ((t || e.f & Io) && e.nodes_start !== null) {
    for (var r = e.nodes_start, i = e.nodes_end; r !== null; ) {
      var o = r === i ? null : (
        /** @type {TemplateNode} */
        /* @__PURE__ */ rn(r)
      );
      r.remove(), r = o;
    }
    n = !0;
  }
  Jr(e, t && !n), Xr(e), Tt(e, 0), me(e, Rt);
  var a = e.transitions;
  if (a !== null)
    for (const c of a)
      c.stop();
  Zr(e);
  var s = e.parent;
  s !== null && s.first !== null && ei(e), e.next = e.prev = e.teardown = e.ctx = e.deps = e.parent = e.fn = e.nodes_start = e.nodes_end = null;
}
function ei(e) {
  var t = e.parent, n = e.prev, r = e.next;
  n !== null && (n.next = r), r !== null && (r.prev = n), t !== null && (t.first === e && (t.first = r), t.last === e && (t.last = n));
}
function Pt(e, t) {
  var n = [];
  Vn(e, n, !0), ti(n, () => {
    ke(e), t && t();
  });
}
function ti(e, t) {
  var n = e.length;
  if (n > 0) {
    var r = () => --n || t();
    for (var i of e)
      i.out(r);
  } else
    t();
}
function Vn(e, t, n) {
  if (!(e.f & xe)) {
    if (e.f ^= xe, e.transitions !== null)
      for (const a of e.transitions)
        (a.is_global || n) && t.push(a);
    for (var r = e.first; r !== null; ) {
      var i = r.next, o = (r.f & st) !== 0 || (r.f & be) !== 0;
      Vn(r, t, o ? n : !1), r = i;
    }
  }
}
function kt(e) {
  ni(e, !0);
}
function ni(e, t) {
  if (e.f & xe) {
    ut(e) && Ut(e), e.f ^= xe;
    for (var n = e.first; n !== null; ) {
      var r = n.next, i = (n.f & st) !== 0 || (n.f & be) !== 0;
      ni(n, i ? t : !1), n = r;
    }
    if (e.transitions !== null)
      for (const o of e.transitions)
        (o.is_global || t) && o.in();
  }
}
let bn = !1, mn = [];
function Fo() {
  bn = !1;
  const e = mn.slice();
  mn = [], hn(e);
}
function ct(e) {
  bn || (bn = !0, queueMicrotask(Fo)), mn.push(e);
}
function en(e) {
  throw new Error("lifecycle_outside_component");
}
let Qt = !1, rt = !1, Gn = !1;
function dr(e) {
  rt = e;
}
function vr(e) {
  Gn = e;
}
let yn = [], Et = 0;
let q = null;
function Fe(e) {
  q = e;
}
let S = null;
function Pe(e) {
  S = e;
}
let Ie = null;
function Ho(e) {
  Ie = e;
}
let ee = null, ve = 0, qe = null;
function Vo(e) {
  qe = e;
}
let ri = 0, nt = !1, L = null;
function ii() {
  return ++ri;
}
function tn() {
  return !ot || L !== null && L.l === null;
}
function ut(e) {
  var a, s;
  var t = e.f;
  if (t & je)
    return !0;
  if (t & at) {
    var n = e.deps, r = (t & We) !== 0;
    if (n !== null) {
      var i;
      if (t & Kt) {
        for (i = 0; i < n.length; i++)
          ((a = n[i]).reactions ?? (a.reactions = [])).push(e);
        e.f ^= Kt;
      }
      for (i = 0; i < n.length; i++) {
        var o = n[i];
        if (ut(
          /** @type {Derived} */
          o
        ) && Qr(
          /** @type {Derived} */
          o
        ), r && S !== null && !nt && !((s = o == null ? void 0 : o.reactions) != null && s.includes(e)) && (o.reactions ?? (o.reactions = [])).push(e), o.version > e.version)
          return !0;
      }
    }
    r || me(e, oe);
  }
  return !1;
}
function Go(e, t, n) {
  throw e;
}
function oi(e) {
  var f;
  var t = ee, n = ve, r = qe, i = q, o = nt, a = Ie, s = L, c = e.f;
  ee = /** @type {null | Value[]} */
  null, ve = 0, qe = null, q = c & (be | St) ? null : e, nt = !rt && (c & We) !== 0, Ie = null, L = e.ctx;
  try {
    var l = (
      /** @type {Function} */
      (0, e.fn)()
    ), d = e.deps;
    if (ee !== null) {
      var u;
      if (Tt(e, ve), d !== null && ve > 0)
        for (d.length = ve + ee.length, u = 0; u < ee.length; u++)
          d[ve + u] = ee[u];
      else
        e.deps = d = ee;
      if (!nt)
        for (u = ve; u < d.length; u++)
          ((f = d[u]).reactions ?? (f.reactions = [])).push(e);
    } else d !== null && ve < d.length && (Tt(e, ve), d.length = ve);
    return l;
  } finally {
    ee = t, ve = n, qe = r, q = i, nt = o, Ie = a, L = s;
  }
}
function zo(e, t) {
  let n = t.reactions;
  if (n !== null) {
    var r = n.indexOf(e);
    if (r !== -1) {
      var i = n.length - 1;
      i === 0 ? n = t.reactions = null : (n[r] = n[i], n.pop());
    }
  }
  n === null && t.f & Re && // Destroying a child effect while updating a parent effect can cause a dependency to appear
  // to be unused, when in fact it is used by the currently-updating parent. Checking `new_deps`
  // allows us to skip the expensive work of disconnecting and immediately reconnecting it
  (ee === null || !ee.includes(t)) && (me(t, at), t.f & (We | Kt) || (t.f ^= Kt), Tt(
    /** @type {Derived} **/
    t,
    0
  ));
}
function Tt(e, t) {
  var n = e.deps;
  if (n !== null)
    for (var r = t; r < n.length; r++)
      zo(e, n[r]);
}
function Ut(e) {
  var t = e.f;
  if (!(t & Rt)) {
    me(e, oe);
    var n = S;
    S = e;
    try {
      t & Cn ? jo(e) : Jr(e), Xr(e), Zr(e);
      var r = oi(e);
      e.teardown = typeof r == "function" ? r : null, e.version = ri;
    } catch (i) {
      Go(
        /** @type {Error} */
        i
      );
    } finally {
      S = n;
    }
  }
}
function Yo() {
  Et > 1e3 && (Et = 0, Oo()), Et++;
}
function $o(e) {
  var t = e.length;
  if (t !== 0) {
    Yo();
    var n = rt;
    rt = !0;
    try {
      for (var r = 0; r < t; r++) {
        var i = e[r];
        i.f & oe || (i.f ^= oe);
        var o = [];
        ai(i, o), Ko(o);
      }
    } finally {
      rt = n;
    }
  }
}
function Ko(e) {
  var t = e.length;
  if (t !== 0)
    for (var n = 0; n < t; n++) {
      var r = e[n];
      !(r.f & (Rt | xe)) && ut(r) && (Ut(r), r.deps === null && r.first === null && r.nodes_start === null && (r.teardown === null ? ei(r) : r.fn = null));
    }
}
function Qo() {
  if (Qt = !1, Et > 1001)
    return;
  const e = yn;
  yn = [], $o(e), Qt || (Et = 0);
}
function nn(e) {
  Qt || (Qt = !0, queueMicrotask(Qo));
  for (var t = e; t.parent !== null; ) {
    t = t.parent;
    var n = t.f;
    if (n & (St | be)) {
      if (!(n & oe)) return;
      t.f ^= oe;
    }
  }
  yn.push(t);
}
function ai(e, t) {
  var n = e.first, r = [];
  e: for (; n !== null; ) {
    var i = n.f, o = (i & be) !== 0, a = o && (i & oe) !== 0;
    if (!a && !(i & xe))
      if (i & Nt) {
        o ? n.f ^= oe : ut(n) && Ut(n);
        var s = n.first;
        if (s !== null) {
          n = s;
          continue;
        }
      } else i & Fr && r.push(n);
    var c = n.next;
    if (c === null) {
      let u = n.parent;
      for (; u !== null; ) {
        if (e === u)
          break e;
        var l = u.next;
        if (l !== null) {
          n = l;
          continue e;
        }
        u = u.parent;
      }
    }
    n = c;
  }
  for (var d = 0; d < r.length; d++)
    s = r[d], t.push(s), ai(s, t);
}
function g(e) {
  var s;
  var t = e.f, n = (t & Re) !== 0;
  if (n && t & Rt) {
    var r = Kr(
      /** @type {Derived} */
      e
    );
    return qn(
      /** @type {Derived} */
      e
    ), r;
  }
  if (q !== null) {
    Ie !== null && Ie.includes(e) && Uo();
    var i = q.deps;
    ee === null && i !== null && i[ve] === e ? ve++ : ee === null ? ee = [e] : ee.push(e), qe !== null && S !== null && S.f & oe && !(S.f & be) && qe.includes(e) && (me(S, je), nn(S));
  } else if (n && /** @type {Derived} */
  e.deps === null) {
    var o = (
      /** @type {Derived} */
      e
    ), a = o.parent;
    a !== null && !((s = a.deriveds) != null && s.includes(o)) && (a.deriveds ?? (a.deriveds = [])).push(o);
  }
  return n && (o = /** @type {Derived} */
  e, ut(o) && Qr(o)), e.v;
}
function Qe(e) {
  const t = q;
  try {
    return q = null, e();
  } finally {
    q = t;
  }
}
const Wo = ~(je | at | oe);
function me(e, t) {
  e.f = e.f & Wo | t;
}
function Zo(e) {
  return (
    /** @type {T} */
    si().get(e)
  );
}
function gr(e, t) {
  return si().set(e, t), t;
}
function si(e) {
  return L === null && en(), L.c ?? (L.c = new Map(Xo(L) || void 0));
}
function Xo(e) {
  let t = e.p;
  for (; t !== null; ) {
    const n = t.c;
    if (n !== null)
      return n;
    t = t.p;
  }
  return null;
}
function Y(e, t = !1, n) {
  L = {
    p: L,
    c: null,
    e: null,
    m: !1,
    s: e,
    x: null,
    l: null
  }, ot && !t && (L.l = {
    s: null,
    u: null,
    r1: [],
    r2: pe(!1)
  });
}
function $(e) {
  const t = L;
  if (t !== null) {
    e !== void 0 && (t.x = e);
    const a = t.e;
    if (a !== null) {
      var n = S, r = q;
      t.e = null;
      try {
        for (var i = 0; i < a.length; i++) {
          var o = a[i];
          Pe(o.effect), Fe(o.reaction), Fn(o.fn);
        }
      } finally {
        Pe(n), Fe(r);
      }
    }
    L = t.p, t.m = !0;
  }
  return e || /** @type {T} */
  {};
}
function J(e) {
  if (!(typeof e != "object" || !e || e instanceof EventTarget)) {
    if (Be in e)
      wn(e);
    else if (!Array.isArray(e))
      for (let t in e) {
        const n = e[t];
        typeof n == "object" && n && Be in n && wn(n);
      }
  }
}
function wn(e, t = /* @__PURE__ */ new Set()) {
  if (typeof e == "object" && e !== null && // We don't want to traverse DOM elements
  !(e instanceof EventTarget) && !t.has(e)) {
    t.add(e), e instanceof Date && e.getTime();
    for (let r in e)
      try {
        wn(e[r], t);
      } catch {
      }
    const n = $t(e);
    if (n !== Object.prototype && n !== Array.prototype && n !== Map.prototype && n !== Set.prototype && n !== Date.prototype) {
      const r = jr(n);
      for (let i in r) {
        const o = r[i].get;
        if (o)
          try {
            o.call(e);
          } catch {
          }
      }
    }
  }
}
function $e(e, t = null, n) {
  if (typeof e != "object" || e === null || Be in e)
    return e;
  const r = $t(e);
  if (r !== mo && r !== yo)
    return e;
  var i = /* @__PURE__ */ new Map(), o = Ot(e), a = pe(0);
  o && i.set("length", pe(
    /** @type {any[]} */
    e.length
  ));
  var s;
  return new Proxy(
    /** @type {any} */
    e,
    {
      defineProperty(c, l, d) {
        (!("value" in d) || d.configurable === !1 || d.enumerable === !1 || d.writable === !1) && Ro();
        var u = i.get(l);
        return u === void 0 ? (u = pe(d.value), i.set(l, u)) : O(u, $e(d.value, s)), !0;
      },
      deleteProperty(c, l) {
        var d = i.get(l);
        if (d === void 0)
          l in c && i.set(l, pe(fe));
        else {
          if (o && typeof l == "string") {
            var u = (
              /** @type {Source<number>} */
              i.get("length")
            ), f = Number(l);
            Number.isInteger(f) && f < u.v && O(u, f);
          }
          O(d, fe), hr(a);
        }
        return !0;
      },
      get(c, l, d) {
        var p;
        if (l === Be)
          return e;
        var u = i.get(l), f = l in c;
        if (u === void 0 && (!f || (p = Se(c, l)) != null && p.writable) && (u = pe($e(f ? c[l] : fe, s)), i.set(l, u)), u !== void 0) {
          var v = g(u);
          return v === fe ? void 0 : v;
        }
        return Reflect.get(c, l, d);
      },
      getOwnPropertyDescriptor(c, l) {
        var d = Reflect.getOwnPropertyDescriptor(c, l);
        if (d && "value" in d) {
          var u = i.get(l);
          u && (d.value = g(u));
        } else if (d === void 0) {
          var f = i.get(l), v = f == null ? void 0 : f.v;
          if (f !== void 0 && v !== fe)
            return {
              enumerable: !0,
              configurable: !0,
              value: v,
              writable: !0
            };
        }
        return d;
      },
      has(c, l) {
        var v;
        if (l === Be)
          return !0;
        var d = i.get(l), u = d !== void 0 && d.v !== fe || Reflect.has(c, l);
        if (d !== void 0 || S !== null && (!u || (v = Se(c, l)) != null && v.writable)) {
          d === void 0 && (d = pe(u ? $e(c[l], s) : fe), i.set(l, d));
          var f = g(d);
          if (f === fe)
            return !1;
        }
        return u;
      },
      set(c, l, d, u) {
        var P;
        var f = i.get(l), v = l in c;
        if (o && l === "length")
          for (var p = d; p < /** @type {Source<number>} */
          f.v; p += 1) {
            var b = i.get(p + "");
            b !== void 0 ? O(b, fe) : p in c && (b = pe(fe), i.set(p + "", b));
          }
        f === void 0 ? (!v || (P = Se(c, l)) != null && P.writable) && (f = pe(void 0), O(f, $e(d, s)), i.set(l, f)) : (v = f.v !== fe, O(f, $e(d, s)));
        var x = Reflect.getOwnPropertyDescriptor(c, l);
        if (x != null && x.set && x.set.call(u, d), !v) {
          if (o && typeof l == "string") {
            var m = (
              /** @type {Source<number>} */
              i.get("length")
            ), E = Number(l);
            Number.isInteger(E) && E >= m.v && O(m, E + 1);
          }
          hr(a);
        }
        return !0;
      },
      ownKeys(c) {
        g(a);
        var l = Reflect.ownKeys(c).filter((f) => {
          var v = i.get(f);
          return v === void 0 || v.v !== fe;
        });
        for (var [d, u] of i)
          u.v !== fe && !(d in c) && l.push(d);
        return l;
      },
      setPrototypeOf() {
        Lo();
      }
    }
  );
}
function hr(e, t = 1) {
  O(e, e.v + t);
}
var pr, li, ci;
function Jo() {
  if (pr === void 0) {
    pr = window;
    var e = Element.prototype, t = Node.prototype;
    li = Se(t, "firstChild").get, ci = Se(t, "nextSibling").get, e.__click = void 0, e.__className = "", e.__attributes = null, e.__styles = null, e.__e = void 0, Text.prototype.__t = void 0;
  }
}
function Dt(e = "") {
  return document.createTextNode(e);
}
// @__NO_SIDE_EFFECTS__
function At(e) {
  return li.call(e);
}
// @__NO_SIDE_EFFECTS__
function rn(e) {
  return ci.call(e);
}
function k(e, t) {
  return /* @__PURE__ */ At(e);
}
function U(e, t) {
  {
    var n = (
      /** @type {DocumentFragment} */
      /* @__PURE__ */ At(
        /** @type {Node} */
        e
      )
    );
    return n instanceof Comment && n.data === "" ? /* @__PURE__ */ rn(n) : n;
  }
}
function N(e, t = 1, n = !1) {
  let r = e;
  for (; t--; )
    r = /** @type {TemplateNode} */
    /* @__PURE__ */ rn(r);
  return r;
}
function ea(e) {
  e.textContent = "";
}
let ui = !1;
function ta(e, t) {
  if (t) {
    const n = document.body;
    e.autofocus = !0, ct(() => {
      document.activeElement === n && e.focus();
    });
  }
}
let _r = !1;
function na() {
  _r || (_r = !0, document.addEventListener(
    "reset",
    (e) => {
      Promise.resolve().then(() => {
        var t;
        if (!e.defaultPrevented)
          for (
            const n of
            /**@type {HTMLFormElement} */
            e.target.elements
          )
            (t = n.__on_r) == null || t.call(n);
      });
    },
    // In the capture phase to guarantee we get noticed of it (no possiblity of stopPropagation)
    { capture: !0 }
  ));
}
function fi(e) {
  var t = q, n = S;
  Fe(null), Pe(null);
  try {
    return e();
  } finally {
    Fe(t), Pe(n);
  }
}
function ra(e, t, n, r = n) {
  e.addEventListener(t, () => fi(n));
  const i = e.__on_r;
  i ? e.__on_r = () => {
    i(), r();
  } : e.__on_r = r, na();
}
const di = /* @__PURE__ */ new Set(), En = /* @__PURE__ */ new Set();
function xn(e, t, n, r) {
  function i(o) {
    if (r.capture || yt.call(t, o), !o.cancelBubble)
      return fi(() => n.call(this, o));
  }
  return e.startsWith("pointer") || e.startsWith("touch") || e === "wheel" ? ct(() => {
    t.addEventListener(e, i, r);
  }) : t.addEventListener(e, i, r), i;
}
function on(e, t, n, r, i) {
  var o = { capture: r, passive: i }, a = xn(e, t, n, o);
  (t === document.body || t === window || t === document) && jn(() => {
    t.removeEventListener(e, a, o);
  });
}
function zn(e) {
  for (var t = 0; t < e.length; t++)
    di.add(e[t]);
  for (var n of En)
    n(e);
}
function yt(e) {
  var E;
  var t = this, n = (
    /** @type {Node} */
    t.ownerDocument
  ), r = e.type, i = ((E = e.composedPath) == null ? void 0 : E.call(e)) || [], o = (
    /** @type {null | Element} */
    i[0] || e.target
  ), a = 0, s = e.__root;
  if (s) {
    var c = i.indexOf(s);
    if (c !== -1 && (t === document || t === /** @type {any} */
    window)) {
      e.__root = t;
      return;
    }
    var l = i.indexOf(t);
    if (l === -1)
      return;
    c <= l && (a = c);
  }
  if (o = /** @type {Element} */
  i[a] || e.target, o !== t) {
    bo(e, "currentTarget", {
      configurable: !0,
      get() {
        return o || n;
      }
    });
    var d = q, u = S;
    Fe(null), Pe(null);
    try {
      for (var f, v = []; o !== null; ) {
        var p = o.assignedSlot || o.parentNode || /** @type {any} */
        o.host || null;
        try {
          var b = o["__" + r];
          if (b !== void 0 && !/** @type {any} */
          o.disabled)
            if (Ot(b)) {
              var [x, ...m] = b;
              x.apply(o, [e, ...m]);
            } else
              b.call(o, e);
        } catch (P) {
          f ? v.push(P) : f = P;
        }
        if (e.cancelBubble || p === t || p === null)
          break;
        o = p;
      }
      if (f) {
        for (let P of v)
          queueMicrotask(() => {
            throw P;
          });
        throw f;
      }
    } finally {
      e.__root = t, delete e.currentTarget, Fe(d), Pe(u);
    }
  }
}
function vi(e) {
  var t = document.createElement("template");
  return t.innerHTML = e, t.content;
}
function it(e, t) {
  var n = (
    /** @type {Effect} */
    S
  );
  n.nodes_start === null && (n.nodes_start = e, n.nodes_end = t);
}
// @__NO_SIDE_EFFECTS__
function I(e, t) {
  var n = (t & vo) !== 0, r = (t & go) !== 0, i, o = !e.startsWith("<!>");
  return () => {
    i === void 0 && (i = vi(o ? e : "<!>" + e), n || (i = /** @type {Node} */
    /* @__PURE__ */ At(i)));
    var a = (
      /** @type {TemplateNode} */
      r ? document.importNode(i, !0) : i.cloneNode(!0)
    );
    if (n) {
      var s = (
        /** @type {TemplateNode} */
        /* @__PURE__ */ At(a)
      ), c = (
        /** @type {TemplateNode} */
        a.lastChild
      );
      it(s, c);
    } else
      it(a, a);
    return a;
  };
}
function He(e = "") {
  {
    var t = Dt(e + "");
    return it(t, t), t;
  }
}
function te() {
  var e = document.createDocumentFragment(), t = document.createComment(""), n = Dt();
  return e.append(t, n), it(t, n), e;
}
function _(e, t) {
  e !== null && e.before(
    /** @type {Node} */
    t
  );
}
function ia(e) {
  return e.endsWith("capture") && e !== "gotpointercapture" && e !== "lostpointercapture";
}
const oa = [
  "beforeinput",
  "click",
  "change",
  "dblclick",
  "contextmenu",
  "focusin",
  "focusout",
  "input",
  "keydown",
  "keyup",
  "mousedown",
  "mousemove",
  "mouseout",
  "mouseover",
  "mouseup",
  "pointerdown",
  "pointermove",
  "pointerout",
  "pointerover",
  "pointerup",
  "touchend",
  "touchmove",
  "touchstart"
];
function aa(e) {
  return oa.includes(e);
}
const sa = {
  // no `class: 'className'` because we handle that separately
  formnovalidate: "formNoValidate",
  ismap: "isMap",
  nomodule: "noModule",
  playsinline: "playsInline",
  readonly: "readOnly"
};
function la(e) {
  return e = e.toLowerCase(), sa[e] ?? e;
}
const ca = ["touchstart", "touchmove"];
function ua(e) {
  return ca.includes(e);
}
function re(e, t) {
  var n = t == null ? "" : typeof t == "object" ? t + "" : t;
  n !== (e.__t ?? (e.__t = e.nodeValue)) && (e.__t = n, e.nodeValue = n == null ? "" : n + "");
}
function br(e, t) {
  return fa(e, t);
}
const Xe = /* @__PURE__ */ new Map();
function fa(e, { target: t, anchor: n, props: r = {}, events: i, context: o, intro: a = !0 }) {
  Jo();
  var s = /* @__PURE__ */ new Set(), c = (u) => {
    for (var f = 0; f < u.length; f++) {
      var v = u[f];
      if (!s.has(v)) {
        s.add(v);
        var p = ua(v);
        t.addEventListener(v, yt, { passive: p });
        var b = Xe.get(v);
        b === void 0 ? (document.addEventListener(v, yt, { passive: p }), Xe.set(v, 1)) : Xe.set(v, b + 1);
      }
    }
  };
  c(Mn(di)), En.add(c);
  var l = void 0, d = qo(() => {
    var u = n ?? t.appendChild(Dt());
    return Le(() => {
      if (o) {
        Y({});
        var f = (
          /** @type {ComponentContext} */
          L
        );
        f.c = o;
      }
      i && (r.$$events = i), l = e(u, r) || {}, o && $();
    }), () => {
      var p;
      for (var f of s) {
        t.removeEventListener(f, yt);
        var v = (
          /** @type {number} */
          Xe.get(f)
        );
        --v === 0 ? (document.removeEventListener(f, yt), Xe.delete(f)) : Xe.set(f, v);
      }
      En.delete(c), mr.delete(l), u !== n && ((p = u.parentNode) == null || p.removeChild(u));
    };
  });
  return mr.set(l, d), l;
}
let mr = /* @__PURE__ */ new WeakMap();
function C(e, t, n, r = null, i = !1) {
  var o = e, a = null, s = null, c = null, l = i ? st : 0;
  Ze(() => {
    c !== (c = !!t()) && (c ? (a ? kt(a) : a = Le(() => n(o)), s && Pt(s, () => {
      s = null;
    })) : (s ? kt(s) : r && (s = Le(() => r(o))), a && Pt(a, () => {
      a = null;
    })));
  }, l);
}
function gi(e, t) {
  return t;
}
function da(e, t, n, r) {
  for (var i = [], o = t.length, a = 0; a < o; a++)
    Vn(t[a].e, i, !0);
  var s = o > 0 && i.length === 0 && n !== null;
  if (s) {
    var c = (
      /** @type {Element} */
      /** @type {Element} */
      n.parentNode
    );
    ea(c), c.append(
      /** @type {Element} */
      n
    ), r.clear(), Me(e, t[0].prev, t[o - 1].next);
  }
  ti(i, () => {
    for (var l = 0; l < o; l++) {
      var d = t[l];
      s || (r.delete(d.k), Me(e, d.prev, d.next)), ke(d.e, !s);
    }
  });
}
function hi(e, t, n, r, i, o = null) {
  var a = e, s = { flags: t, items: /* @__PURE__ */ new Map(), first: null }, c = (t & qr) !== 0;
  if (c) {
    var l = (
      /** @type {Element} */
      e
    );
    a = l.appendChild(Dt());
  }
  var d = null, u = !1;
  Ze(() => {
    var f = n(), v = Ot(f) ? f : f == null ? [] : Mn(f), p = v.length;
    if (!(u && p === 0)) {
      u = p === 0;
      {
        var b = (
          /** @type {Effect} */
          q
        );
        va(v, s, a, i, t, (b.f & xe) !== 0, r);
      }
      o !== null && (p === 0 ? d ? kt(d) : d = Le(() => o(a)) : d !== null && Pt(d, () => {
        d = null;
      })), n();
    }
  });
}
function va(e, t, n, r, i, o, a) {
  var X, Te, Ue, Ae;
  var s = (i & oo) !== 0, c = (i & (Ln | Un)) !== 0, l = e.length, d = t.items, u = t.first, f = u, v, p = null, b, x = [], m = [], E, P, h, w;
  if (s)
    for (w = 0; w < l; w += 1)
      E = e[w], P = a(E, w), h = d.get(P), h !== void 0 && ((X = h.a) == null || X.measure(), (b ?? (b = /* @__PURE__ */ new Set())).add(h));
  for (w = 0; w < l; w += 1) {
    if (E = e[w], P = a(E, w), h = d.get(P), h === void 0) {
      var M = f ? (
        /** @type {TemplateNode} */
        f.e.nodes_start
      ) : n;
      p = ha(
        M,
        t,
        p,
        p === null ? t.first : p.next,
        E,
        P,
        w,
        r,
        i
      ), d.set(P, p), x = [], m = [], f = p.next;
      continue;
    }
    if (c && ga(h, E, w, i), h.e.f & xe && (kt(h.e), s && ((Te = h.a) == null || Te.unfix(), (b ?? (b = /* @__PURE__ */ new Set())).delete(h))), h !== f) {
      if (v !== void 0 && v.has(h)) {
        if (x.length < m.length) {
          var j = m[0], T;
          p = j.prev;
          var H = x[0], z = x[x.length - 1];
          for (T = 0; T < x.length; T += 1)
            yr(x[T], j, n);
          for (T = 0; T < m.length; T += 1)
            v.delete(m[T]);
          Me(t, H.prev, z.next), Me(t, p, H), Me(t, z, j), f = j, p = z, w -= 1, x = [], m = [];
        } else
          v.delete(h), yr(h, f, n), Me(t, h.prev, h.next), Me(t, h, p === null ? t.first : p.next), Me(t, p, h), p = h;
        continue;
      }
      for (x = [], m = []; f !== null && f.k !== P; )
        (o || !(f.e.f & xe)) && (v ?? (v = /* @__PURE__ */ new Set())).add(f), m.push(f), f = f.next;
      if (f === null)
        continue;
      h = f;
    }
    x.push(h), p = h, f = h.next;
  }
  if (f !== null || v !== void 0) {
    for (var ce = v === void 0 ? [] : Mn(v); f !== null; )
      (o || !(f.e.f & xe)) && ce.push(f), f = f.next;
    var A = ce.length;
    if (A > 0) {
      var V = i & qr && l === 0 ? n : null;
      if (s) {
        for (w = 0; w < A; w += 1)
          (Ue = ce[w].a) == null || Ue.measure();
        for (w = 0; w < A; w += 1)
          (Ae = ce[w].a) == null || Ae.fix();
      }
      da(t, ce, V, d);
    }
  }
  s && ct(() => {
    var Oe;
    if (b !== void 0)
      for (h of b)
        (Oe = h.a) == null || Oe.apply();
  }), S.first = t.first && t.first.e, S.last = p && p.e;
}
function ga(e, t, n, r) {
  r & Ln && pn(e.v, t), r & Un ? pn(
    /** @type {Value<number>} */
    e.i,
    n
  ) : e.i = n;
}
function ha(e, t, n, r, i, o, a, s, c) {
  var l = (c & Ln) !== 0, d = (c & ao) === 0, u = l ? d ? /* @__PURE__ */ Jt(i) : pe(i) : i, f = c & Un ? pe(a) : a, v = {
    i: f,
    v: u,
    k: o,
    a: null,
    // @ts-expect-error
    e: null,
    prev: n,
    next: r
  };
  try {
    return v.e = Le(() => s(e, u, f), ui), v.e.prev = n && n.e, v.e.next = r && r.e, n === null ? t.first = v : (n.next = v, n.e.next = v.e), r !== null && (r.prev = v, r.e.prev = v.e), v;
  } finally {
  }
}
function yr(e, t, n) {
  for (var r = e.next ? (
    /** @type {TemplateNode} */
    e.next.e.nodes_start
  ) : n, i = t ? (
    /** @type {TemplateNode} */
    t.e.nodes_start
  ) : n, o = (
    /** @type {TemplateNode} */
    e.e.nodes_start
  ); o !== r; ) {
    var a = (
      /** @type {TemplateNode} */
      /* @__PURE__ */ rn(o)
    );
    i.before(o), o = a;
  }
}
function Me(e, t, n) {
  t === null ? e.first = n : (t.next = n, t.e.next = n && n.e), n !== null && (n.prev = t, n.e.prev = t && t.e);
}
function Mt(e, t, n, r, i) {
  var o = e, a = "", s;
  Ze(() => {
    a !== (a = t() ?? "") && (s !== void 0 && (ke(s), s = void 0), a !== "" && (s = Le(() => {
      var c = a + "", l = vi(c);
      it(
        /** @type {TemplateNode} */
        /* @__PURE__ */ At(l),
        /** @type {TemplateNode} */
        l.lastChild
      ), o.before(l);
    })));
  });
}
function W(e, t, n, r, i) {
  var s;
  var o = (s = t.$$slots) == null ? void 0 : s[n], a = !1;
  o === !0 && (o = t[n === "default" ? "children" : n], a = !0), o === void 0 || o(e, a ? () => r : r);
}
function pa(e, t, ...n) {
  var r = e, i = Ce, o;
  Ze(() => {
    i !== (i = t()) && (o && (ke(o), o = null), o = Le(() => (
      /** @type {SnippetFn} */
      i(r, ...n)
    )));
  }, st);
}
function _a(e, t, n) {
  var r = e, i, o;
  Ze(() => {
    i !== (i = t()) && (o && (Pt(o), o = null), i && (o = Le(() => n(r, i))));
  }, st);
}
function pi(e, t, n, r, i, o) {
  var a, s, c = null, l = (
    /** @type {TemplateNode} */
    e
  ), d;
  Ze(() => {
    const u = t() || null;
    var f = u === "svg" ? Dn : null;
    u !== a && (d && (u === null ? Pt(d, () => {
      d = null, s = null;
    }) : u === s ? kt(d) : ke(d)), u && u !== s && (d = Le(() => {
      if (c = f ? document.createElementNS(f, u) : document.createElement(u), it(c, c), r) {
        var v = (
          /** @type {TemplateNode} */
          c.appendChild(Dt())
        );
        r(c, v);
      }
      S.nodes_end = c, l.before(c);
    })), a = u, a && (s = a));
  }, st);
}
function D(e, t) {
  ct(() => {
    var n = e.getRootNode(), r = (
      /** @type {ShadowRoot} */
      n.host ? (
        /** @type {ShadowRoot} */
        n
      ) : (
        /** @type {Document} */
        n.head ?? /** @type {Document} */
        n.ownerDocument.head
      )
    );
    if (!r.querySelector("#" + t.hash)) {
      const i = document.createElement("style");
      i.id = t.hash, i.textContent = t.code, r.appendChild(i);
    }
  });
}
function G(e, t, n, r) {
  var i = e.__attributes ?? (e.__attributes = {});
  i[t] !== (i[t] = n) && (t === "style" && "__styles" in e && (e.__styles = {}), t === "loading" && (e[Po] = n), n == null ? e.removeAttribute(t) : typeof n != "string" && bi(e).includes(t) ? e[t] = n : e.setAttribute(t, n));
}
function _i(e, t, n, r, i = !1, o = !1, a = !1) {
  var s = t || {}, c = e.tagName === "OPTION";
  for (var l in t)
    l in n || (n[l] = null);
  r !== void 0 && (n.class = n.class ? n.class + " " + r : r);
  var d = bi(e), u = (
    /** @type {Record<string, unknown>} **/
    e.__attributes ?? (e.__attributes = {})
  ), f = [];
  for (const m in n) {
    let E = n[m];
    if (c && m === "value" && E == null) {
      e.value = e.__value = "", s[m] = E;
      continue;
    }
    var v = s[m];
    if (E !== v) {
      s[m] = E;
      var p = m[0] + m[1];
      if (p !== "$$") {
        if (p === "on") {
          const P = {}, h = "$$" + m;
          let w = m.slice(2);
          var b = aa(w);
          if (ia(w) && (w = w.slice(0, -7), P.capture = !0), !b && v) {
            if (E != null) continue;
            e.removeEventListener(w, s[h], P), s[h] = null;
          }
          if (E != null)
            if (b)
              e[`__${w}`] = E, zn([w]);
            else {
              let M = function(j) {
                s[m].call(this, j);
              };
              t ? s[h] = xn(w, e, M, P) : f.push([
                m,
                E,
                () => s[h] = xn(w, e, M, P)
              ]);
            }
        } else if (m === "style" && E != null)
          e.style.cssText = E + "";
        else if (m === "autofocus")
          ta(
            /** @type {HTMLElement} */
            e,
            !!E
          );
        else if (m === "__value" || m === "value" && E != null)
          e.value = e[m] = e.__value = E;
        else {
          var x = m;
          i || (x = la(x)), E == null && !o ? (u[m] = null, e.removeAttribute(m)) : d.includes(x) && (o || typeof E != "string") ? e[x] = E : typeof E != "function" && G(e, x, E);
        }
        m === "style" && "__styles" in e && (e.__styles = {});
      }
    }
  }
  return t || ct(() => {
    if (e.isConnected)
      for (const [m, E, P] of f)
        s[m] === E && P();
  }), s;
}
var wr = /* @__PURE__ */ new Map();
function bi(e) {
  var t = wr.get(e.nodeName);
  if (t) return t;
  wr.set(e.nodeName, t = []);
  for (var n, r = $t(e), i = Element.prototype; i !== r; ) {
    n = jr(r);
    for (var o in n)
      n[o].set && t.push(o);
    r = $t(r);
  }
  return t;
}
function Yn(e, t) {
  var n = e.__className, r = ba(t);
  (n !== r || ui) && (t == null ? e.removeAttribute("class") : e.className = r, e.__className = r);
}
function ba(e) {
  return e ?? "";
}
function ma(e, t, n) {
  if (n) {
    if (e.classList.contains(t)) return;
    e.classList.add(t);
  } else {
    if (!e.classList.contains(t)) return;
    e.classList.remove(t);
  }
}
function ya(e, t, n = t) {
  var r = tn();
  ra(e, "input", () => {
    var i = Er(e) ? xr(e.value) : e.value;
    n(i), r && i !== (i = t()) && (e.value = i ?? "");
  }), Lt(() => {
    var i = t();
    Er(e) && i === xr(e.value) || e.type === "date" && !i && !e.value || i !== e.value && (e.value = i ?? "");
  });
}
function Er(e) {
  var t = e.type;
  return t === "number" || t === "range";
}
function xr(e) {
  return e === "" ? null : +e;
}
function Ft(e, t, n) {
  var r = Se(e, t);
  r && r.set && (e[t] = n, jn(() => {
    e[t] = null;
  }));
}
function Ir(e, t) {
  return e === t || (e == null ? void 0 : e[Be]) === t;
}
function wa(e = {}, t, n, r) {
  return Fn(() => {
    var i, o;
    return Lt(() => {
      i = o, o = [], Qe(() => {
        e !== n(...o) && (t(e, ...o), i && Ir(n(...i), e) && t(null, ...i));
      });
    }), () => {
      ct(() => {
        o && Ir(n(...o), e) && t(null, ...o);
      });
    };
  }), e;
}
function mi(e) {
  return function(...t) {
    var n = (
      /** @type {Event} */
      t[0]
    );
    return n.preventDefault(), e == null ? void 0 : e.apply(this, t);
  };
}
function ye(e = !1) {
  const t = (
    /** @type {ComponentContextLegacy} */
    L
  ), n = t.l.u;
  if (!n) return;
  let r = () => J(t.s);
  if (e) {
    let i = 0, o = (
      /** @type {Record<string, any>} */
      {}
    );
    const a = /* @__PURE__ */ K(() => {
      let s = !1;
      const c = t.s;
      for (const l in c)
        c[l] !== o[l] && (o[l] = c[l], s = !0);
      return s && i++, i;
    });
    r = () => g(a);
  }
  n.b.length && Bo(() => {
    Pr(t, r), hn(n.b);
  }), _n(() => {
    const i = Qe(() => n.m.map(wo));
    return () => {
      for (const o of i)
        typeof o == "function" && o();
    };
  }), n.a.length && _n(() => {
    Pr(t, r), hn(n.a);
  });
}
function Pr(e, t) {
  if (e.l.s)
    for (const n of e.l.s) g(n);
  t();
}
function yi(e, t) {
  var o;
  var n = (
    /** @type {Record<string, Function[] | Function>} */
    (o = e.$$events) == null ? void 0 : o[t.type]
  ), r = Ot(n) ? n.slice() : n == null ? [] : [n];
  for (var i of r)
    i.call(this, t);
}
function an(e) {
  L === null && en(), ot && L.l !== null ? wi(L).m.push(e) : _n(() => {
    const t = Qe(e);
    if (typeof t == "function") return (
      /** @type {() => void} */
      t
    );
  });
}
function Ea(e, t, { bubbles: n = !1, cancelable: r = !1 } = {}) {
  return new CustomEvent(e, { detail: t, bubbles: n, cancelable: r });
}
function xa() {
  const e = L;
  return e === null && en(), (t, n, r) => {
    var o;
    const i = (
      /** @type {Record<string, Function | Function[]>} */
      (o = e.s.$$events) == null ? void 0 : o[
        /** @type {any} */
        t
      ]
    );
    if (i) {
      const a = Ot(i) ? i.slice() : [i], s = Ea(
        /** @type {string} */
        t,
        n,
        r
      );
      for (const c of a)
        c.call(e.x, s);
      return !s.defaultPrevented;
    }
    return !0;
  };
}
function Ia(e) {
  L === null && en(), L.l === null && No(), wi(L).b.push(e);
}
function wi(e) {
  var t = (
    /** @type {ComponentContextLegacy} */
    e.l
  );
  return t.u ?? (t.u = { a: [], b: [], m: [] });
}
function Pa(e, t, n) {
  if (e == null)
    return t(void 0), Ce;
  const r = Qe(
    () => e.subscribe(
      t,
      // @ts-expect-error
      n
    )
  );
  return r.unsubscribe ? () => r.unsubscribe() : r;
}
let Ht = !1;
function ka(e, t, n) {
  const r = n[t] ?? (n[t] = {
    store: null,
    source: /* @__PURE__ */ Jt(void 0),
    unsubscribe: Ce
  });
  if (r.store !== e)
    if (r.unsubscribe(), r.store = e ?? null, e == null)
      r.source.v = void 0, r.unsubscribe = Ce;
    else {
      var i = !0;
      r.unsubscribe = Pa(e, (o) => {
        i ? r.source.v = o : O(r.source, o);
      }), i = !1;
    }
  return g(r.source);
}
function Ta() {
  const e = {};
  return jn(() => {
    for (var t in e)
      e[t].unsubscribe();
  }), e;
}
function Aa(e) {
  var t = Ht;
  try {
    return Ht = !1, [e(), Ht];
  } finally {
    Ht = t;
  }
}
const Oa = {
  get(e, t) {
    if (!e.exclude.includes(t))
      return e.props[t];
  },
  set(e, t) {
    return !1;
  },
  getOwnPropertyDescriptor(e, t) {
    if (!e.exclude.includes(t) && t in e.props)
      return {
        enumerable: !0,
        configurable: !0,
        value: e.props[t]
      };
  },
  has(e, t) {
    return e.exclude.includes(t) ? !1 : t in e.props;
  },
  ownKeys(e) {
    return Reflect.ownKeys(e.props).filter((t) => !e.exclude.includes(t));
  }
};
// @__NO_SIDE_EFFECTS__
function sn(e, t, n) {
  return new Proxy(
    { props: e, exclude: t },
    Oa
  );
}
const Na = {
  get(e, t) {
    let n = e.props.length;
    for (; n--; ) {
      let r = e.props[n];
      if (_t(r) && (r = r()), typeof r == "object" && r !== null && t in r) return r[t];
    }
  },
  set(e, t, n) {
    let r = e.props.length;
    for (; r--; ) {
      let i = e.props[r];
      _t(i) && (i = i());
      const o = Se(i, t);
      if (o && o.set)
        return o.set(n), !0;
    }
    return !1;
  },
  getOwnPropertyDescriptor(e, t) {
    let n = e.props.length;
    for (; n--; ) {
      let r = e.props[n];
      if (_t(r) && (r = r()), typeof r == "object" && r !== null && t in r) {
        const i = Se(r, t);
        return i && !i.configurable && (i.configurable = !0), i;
      }
    }
  },
  has(e, t) {
    if (t === Be || t === Vr) return !1;
    for (let n of e.props)
      if (_t(n) && (n = n()), n != null && t in n) return !0;
    return !1;
  },
  ownKeys(e) {
    const t = [];
    for (let n of e.props) {
      _t(n) && (n = n());
      for (const r in n)
        t.includes(r) || t.push(r);
    }
    return t;
  }
};
function Ct(...e) {
  return new Proxy({ props: e }, Na);
}
function kr(e) {
  for (var t = S, n = S; t !== null && !(t.f & (be | St)); )
    t = t.parent;
  try {
    return Pe(t), e();
  } finally {
    Pe(n);
  }
}
function y(e, t, n, r) {
  var j;
  var i = (n & so) !== 0, o = !ot || (n & lo) !== 0, a = (n & uo) !== 0, s = (n & fo) !== 0, c = !1, l;
  a ? [l, c] = Aa(() => (
    /** @type {V} */
    e[t]
  )) : l = /** @type {V} */
  e[t];
  var d = Be in e || Vr in e, u = ((j = Se(e, t)) == null ? void 0 : j.set) ?? (d && a && t in e ? (T) => e[t] = T : void 0), f = (
    /** @type {V} */
    r
  ), v = !0, p = !1, b = () => (p = !0, v && (v = !1, s ? f = Qe(
    /** @type {() => V} */
    r
  ) : f = /** @type {V} */
  r), f);
  l === void 0 && r !== void 0 && (u && o && So(), l = b(), u && u(l));
  var x;
  if (o)
    x = () => {
      var T = (
        /** @type {V} */
        e[t]
      );
      return T === void 0 ? b() : (v = !0, p = !1, T);
    };
  else {
    var m = kr(
      () => (i ? K : ge)(() => (
        /** @type {V} */
        e[t]
      ))
    );
    m.f |= xo, x = () => {
      var T = g(m);
      return T !== void 0 && (f = /** @type {V} */
      void 0), T === void 0 ? f : T;
    };
  }
  if (!(n & co))
    return x;
  if (u) {
    var E = e.$$legacy;
    return function(T, H) {
      return arguments.length > 0 ? ((!o || !H || E || c) && u(H ? x() : T), T) : x();
    };
  }
  var P = !1, h = !1, w = /* @__PURE__ */ Jt(l), M = kr(
    () => /* @__PURE__ */ K(() => {
      var T = x(), H = g(w);
      return P ? (P = !1, h = !0, H) : (h = !1, w.v = T);
    })
  );
  return i || (M.equals = Bn), function(T, H) {
    if (arguments.length > 0) {
      const z = H ? g(M) : o && a ? $e(T) : T;
      return M.equals(z) || (P = !0, O(w, z), p && f !== void 0 && (f = z), Qe(() => g(M))), T;
    }
    return g(M);
  };
}
const Tr = {
  translate: /* @__PURE__ */ new Map([
    [!0, "yes"],
    [!1, "no"]
  ])
};
function Sa(e, t, n = !1) {
  const r = e in Tr && Tr[e].get(t) || t, i = n ? "" : `="${po(r)}"`;
  return ` ${e}${i}`;
}
const Ra = {
  hash: "svelte-1bzqb13",
  code: ".rcb-modal-section.svelte-1bzqb13 {padding:8px 0px;display:flex;}section.rcb-modal-section.svelte-1bzqb13 {flex-grow:1;}.rcb-modal-section.svelte-1bzqb13:last-of-type {padding:0;}"
};
function ft(e, t) {
  D(e, Ra);
  let n = y(t, "as", 8, "section");
  var r = te(), i = U(r);
  pi(i, n, !1, (o, a) => {
    _i(o, null, { class: "rcb-modal-section" }, "svelte-1bzqb13", o.namespaceURI === Dn, o.nodeName.includes("-"));
    var s = te(), c = U(s);
    W(c, t, "default", {}), _(a, s);
  }), _(e, r);
}
var de = /* @__PURE__ */ ((e) => (e[e.Silent = 0] = "Silent", e[e.Error = 1] = "Error", e[e.Warn = 2] = "Warn", e[e.Info = 3] = "Info", e[e.Debug = 4] = "Debug", e[e.Verbose = 5] = "Verbose", e))(de || {});
class F {
  static setLogLevel(t) {
    this.logLevel = t;
  }
  static log(t, n = this.logLevel) {
    const r = `[Purchases] ${t}`;
    if (!(this.logLevel < n || n === de.Silent))
      switch (n) {
        case de.Error:
          console.error(r);
          break;
        case de.Warn:
          console.warn(r);
          break;
        case de.Info:
          console.info(r);
          break;
        case de.Debug:
          console.debug(r);
          break;
        case de.Verbose:
          console.debug(r);
          break;
      }
  }
  static errorLog(t) {
    this.log(t, de.Error);
  }
  static warnLog(t) {
    this.log(t, de.Warn);
  }
  static infoLog(t) {
    this.log(t, de.Info);
  }
  static debugLog(t) {
    this.log(t, de.Debug);
  }
  static verboseLog(t) {
    this.log(t, de.Verbose);
  }
}
R(F, "logLevel", de.Silent);
var Ne = /* @__PURE__ */ ((e) => (e.Year = "year", e.Month = "month", e.Week = "week", e.Day = "day", e))(Ne || {});
function $n(e) {
  const t = e.match(/^PT?([0-9]+)([MDYW])$/);
  if (!t || t.length < 3)
    return F.errorLog(`Invalid ISO 8601 duration format: ${e}`), null;
  const n = parseInt(t[1]);
  switch (t[2]) {
    case "Y":
      return {
        number: n,
        unit: "year"
        /* Year */
      };
    case "M":
      return {
        number: n,
        unit: "month"
        /* Month */
      };
    case "W":
      return {
        number: n,
        unit: "week"
        /* Week */
      };
    case "D":
      return {
        number: n,
        unit: "day"
        /* Day */
      };
    default:
      return F.errorLog(`Invalid ISO 8601 unit duration format: ${e}`), null;
  }
}
const La = (e, t, n) => {
  const r = e / 1e6;
  return new Intl.NumberFormat(n, {
    style: "currency",
    currency: t
  }).format(r);
}, Ua = (e) => {
  const t = e.number;
  if (t === 1)
    switch (e.unit) {
      case Ne.Year:
        return "yearly";
      case Ne.Month:
        return "monthly";
      case Ne.Week:
        return "weekly";
      case Ne.Day:
        return "daily";
    }
  else
    return `every ${t} ${e.unit}s`;
}, Da = (e) => {
  const t = e.number;
  if (t === 1)
    switch (e.unit) {
      case Ne.Year:
        return "1 year";
      case Ne.Month:
        return "1 month";
      case Ne.Week:
        return "1 week";
      case Ne.Day:
        return "1 day";
    }
  else
    return `${t} ${e.unit}s`;
}, Ma = (e) => {
  const t = $n(e);
  return t ? Ua(t) : "unknown";
}, Ca = (e) => {
  const t = $n(e);
  return t ? Da(t) : "unknown";
};
function Ba(e) {
  return e != null;
}
var qa = /* @__PURE__ */ ((e) => (e.Unknown = "unknown", e.Custom = "custom", e.Lifetime = "$rc_lifetime", e.Annual = "$rc_annual", e.SixMonth = "$rc_six_month", e.ThreeMonth = "$rc_three_month", e.TwoMonth = "$rc_two_month", e.Monthly = "$rc_monthly", e.Weekly = "$rc_weekly", e))(qa || {}), Bt = /* @__PURE__ */ ((e) => (e.Subscription = "subscription", e.Consumable = "consumable", e.NonConsumable = "non_consumable", e))(Bt || {});
const Ei = (e) => ({
  amount: e.amount_micros / 1e4,
  amountMicros: e.amount_micros,
  currency: e.currency,
  formattedPrice: La(e.amount_micros, e.currency)
}), Ar = (e) => {
  const t = e.period_duration;
  return {
    periodDuration: t,
    period: t ? $n(t) : null,
    cycleCount: e.cycle_count,
    price: e.price ? Ei(e.price) : null
  };
}, ja = (e) => e.base == null ? (F.debugLog("Missing base phase for subscription option. Ignoring."), null) : {
  id: e.id,
  priceId: e.price_id,
  base: Ar(e.base),
  trial: e.trial ? Ar(e.trial) : null
}, Fa = (e) => e.base_price == null ? (F.debugLog(
  "Missing base price for non-subscription option. Ignoring."
), null) : {
  id: e.id,
  priceId: e.price_id,
  basePrice: Ei(e.base_price)
}, Ha = (e, t) => {
  const n = e.product_type;
  return n === "subscription" ? Ga(
    e,
    t,
    n
  ) : Va(
    e,
    t,
    n
  );
}, Va = (e, t, n) => {
  const r = {};
  if (Object.entries(e.purchase_options).forEach(
    ([a, s]) => {
      const c = Fa(
        s
      );
      c != null && (r[a] = c);
    }
  ), Object.keys(r).length === 0)
    return F.debugLog(
      `Product ${e.identifier} has no purchase options. Ignoring.`
    ), null;
  const i = e.default_purchase_option_id, o = i && i in e.purchase_options ? r[i] : null;
  return o == null ? (F.debugLog(
    `Product ${e.identifier} has no default purchase option. Ignoring.`
  ), null) : {
    identifier: e.identifier,
    displayName: e.title,
    title: e.title,
    description: e.description,
    productType: n,
    currentPrice: o.basePrice,
    normalPeriodDuration: null,
    presentedOfferingIdentifier: t.offeringIdentifier,
    presentedOfferingContext: t,
    defaultPurchaseOption: o,
    defaultSubscriptionOption: null,
    subscriptionOptions: {},
    defaultNonSubscriptionOption: o
  };
}, Ga = (e, t, n) => {
  const r = {};
  if (Object.entries(e.purchase_options).forEach(
    ([s, c]) => {
      const l = ja(c);
      l != null && (r[s] = l);
    }
  ), Object.keys(r).length === 0)
    return F.debugLog(
      `Product ${e.identifier} has no subscription options. Ignoring.`
    ), null;
  const i = e.default_purchase_option_id, o = i && i in r ? r[i] : null;
  if (o == null)
    return F.debugLog(
      `Product ${e.identifier} has no default subscription option. Ignoring.`
    ), null;
  const a = o.base.price;
  return a == null ? (F.debugLog(
    `Product ${e.identifier} default option has no base price. Ignoring.`
  ), null) : {
    identifier: e.identifier,
    displayName: e.title,
    title: e.title,
    description: e.description,
    productType: n,
    currentPrice: a,
    normalPeriodDuration: o.base.periodDuration,
    presentedOfferingIdentifier: t.offeringIdentifier,
    presentedOfferingContext: t,
    defaultPurchaseOption: o,
    defaultSubscriptionOption: o,
    subscriptionOptions: r,
    defaultNonSubscriptionOption: null
  };
}, za = (e, t, n) => {
  const r = n[t.platform_product_identifier];
  if (r === void 0) return null;
  const i = Ha(r, e);
  return i === null ? null : {
    identifier: t.identifier,
    rcBillingProduct: i,
    packageType: $a(t.identifier)
  };
}, Ya = (e, t, n, r) => {
  const i = {
    offeringIdentifier: t.identifier,
    targetingContext: e && r ? {
      ruleId: r.rule_id,
      revision: r.revision
    } : null,
    placementIdentifier: null
  }, o = t.packages.map(
    (s) => za(i, s, n)
  ).filter(Ba), a = {};
  for (const s of o)
    s != null && (a[s.identifier] = s);
  return o.length == 0 ? null : {
    identifier: t.identifier,
    serverDescription: t.description,
    metadata: t.metadata,
    packagesById: a,
    availablePackages: o,
    lifetime: a.$rc_lifetime ?? null,
    annual: a.$rc_annual ?? null,
    sixMonth: a.$rc_six_month ?? null,
    threeMonth: a.$rc_three_month ?? null,
    twoMonth: a.$rc_two_month ?? null,
    monthly: a.$rc_monthly ?? null,
    weekly: a.$rc_weekly ?? null,
    paywall_components: t.paywall_components ? t.paywall_components : null
  };
};
function $a(e) {
  switch (e) {
    case "$rc_lifetime":
      return "$rc_lifetime";
    case "$rc_annual":
      return "$rc_annual";
    case "$rc_six_month":
      return "$rc_six_month";
    case "$rc_three_month":
      return "$rc_three_month";
    case "$rc_two_month":
      return "$rc_two_month";
    case "$rc_monthly":
      return "$rc_monthly";
    case "$rc_weekly":
      return "$rc_weekly";
    default:
      return e.startsWith("$rc_") ? "unknown" : "custom";
  }
}
var Ka = /* @__PURE__ */ I('<span class="rcb-product-price-after-trial svelte-1mdrqnh"> </span>'), Qa = /* @__PURE__ */ I('<span class="rcb-product-description svelte-1mdrqnh"> </span>'), Wa = /* @__PURE__ */ I("<li> </li>"), Za = /* @__PURE__ */ I('<span class="rcb-product-price svelte-1mdrqnh"><!> <!></span> <!> <!> <ul class="rcb-product-details svelte-1mdrqnh"><!> <li>Continues until canceled</li> <li>Cancel anytime</li></ul>', 1), Xa = /* @__PURE__ */ I('<span class="rcb-product-description svelte-1mdrqnh"> </span>'), Ja = /* @__PURE__ */ I('<span class="rcb-product-price svelte-1mdrqnh"> </span> <!>', 1), es = /* @__PURE__ */ I('<div class="rcb-pricing-info svelte-1mdrqnh"><span class="rc-product-title"> </span> <!> <!></div>');
const ts = {
  hash: "svelte-1mdrqnh",
  code: `.rcb-pricing-info.svelte-1mdrqnh {display:flex;flex-direction:column;margin-top:102px;font-weight:500;}.rcb-product-price.svelte-1mdrqnh {color:var(--rc-color-grey-text-dark);font-size:24px;margin:12px 0px;}.rcb-product-description.svelte-1mdrqnh {color:var(--rc-color-grey-text-dark);margin:0 0 12px 0;}.rcb-product-price-after-trial.svelte-1mdrqnh {margin-bottom:12px;}.rcb-product-details.svelte-1mdrqnh {color:var(--rc-color-grey-text-light);list-style-type:disc;list-style-position:inside;margin:0px;padding:0px;}

    @media screen and (max-width: 960px) {.rcb-pricing-info.svelte-1mdrqnh {margin-top:48px;}
    }`
};
function Or(e, t) {
  var u;
  Y(t, !1), D(e, ts);
  let n = y(t, "productDetails", 8), r = y(t, "purchaseOption", 8), i = y(t, "brandingAppearance", 8, void 0);
  const o = n().productType === Bt.Subscription, a = r(), s = r(), c = a == null ? void 0 : a.trial, l = (u = a == null ? void 0 : a.base) == null ? void 0 : u.price, d = s == null ? void 0 : s.basePrice;
  ye(), ft(e, {
    children: (f, v) => {
      var p = es(), b = k(p), x = k(b), m = N(b, 2);
      C(m, () => o, (P) => {
        var h = Za(), w = U(h), M = k(w);
        C(M, () => c == null ? void 0 : c.periodDuration, (A) => {
          var V = He();
          B(() => re(V, `${Ca(c.periodDuration) ?? ""} free trial`)), _(A, V);
        });
        var j = N(M, 2);
        C(j, () => !(c != null && c.periodDuration) && l, (A) => {
          var V = He();
          B(() => re(V, l.formattedPrice)), _(A, V);
        });
        var T = N(w, 2);
        C(T, () => c && l, (A) => {
          var V = Ka(), X = k(V);
          B(() => re(X, c && l && `${l.formattedPrice} after end of trial`)), _(A, V);
        });
        var H = N(T, 2);
        C(H, () => {
          var A;
          return ((A = i()) == null ? void 0 : A.show_product_description) && n().description;
        }, (A) => {
          var V = Qa(), X = k(V);
          B(() => re(X, n().description)), _(A, V);
        });
        var z = N(H, 2), ce = k(z);
        C(ce, () => n().normalPeriodDuration, (A) => {
          var V = Wa(), X = k(V);
          B(() => re(X, `Renews ${Ma(n().normalPeriodDuration) ?? ""}`)), _(A, V);
        }), _(P, h);
      });
      var E = N(m, 2);
      C(E, () => !o, (P) => {
        var h = Ja(), w = U(h), M = k(w), j = N(w, 2);
        C(j, () => {
          var T;
          return (T = i()) == null ? void 0 : T.show_product_description;
        }, (T) => {
          var H = Xa(), z = k(H);
          B(() => re(z, n().description)), _(T, H);
        }), B(() => re(M, d == null ? void 0 : d.formattedPrice)), _(P, h);
      }), B(() => re(x, n().title)), _(f, p);
    },
    $$slots: { default: !0 }
  }), $();
}
const ns = '<svg width="80" height="80" viewBox="0 0 24 24" fill="currentColor" xmlns="http://www.w3.org/2000/svg" transform="rotate(0 0 0)"><path d="M12 3.5C7.30558 3.5 3.5 7.30558 3.5 12C3.5 16.6944 7.30558 20.5 12 20.5C16.6944 20.5 20.5 16.6944 20.5 12C20.5 11.5858 20.8358 11.25 21.25 11.25C21.6642 11.25 22 11.5858 22 12C22 17.5228 17.5228 22 12 22C6.47715 22 2 17.5228 2 12C2 6.47715 6.47715 2 12 2C12.4142 2 12.75 2.33579 12.75 2.75C12.75 3.16421 12.4142 3.5 12 3.5Z" fill="currentColor"/></svg>';
var rs = /* @__PURE__ */ I('<div style="color:var(--rc-color-accent);" class="rcb-ui-asset-icon svelte-16h2a8k"><!></div>');
const is = {
  hash: "svelte-16h2a8k",
  code: `
    @-webkit-keyframes svelte-16h2a8k-rotating /* Safari and Chrome */ {
        from {
            -webkit-transform: rotate(0deg);
            -o-transform: rotate(0deg);
            transform: rotate(0deg);
        }
        to {
            -webkit-transform: rotate(360deg);
            -o-transform: rotate(360deg);
            transform: rotate(360deg);
        }
    }
    @keyframes svelte-16h2a8k-rotating {
        from {
            -ms-transform: rotate(0deg);
            -moz-transform: rotate(0deg);
            -webkit-transform: rotate(0deg);
            -o-transform: rotate(0deg);
            transform: rotate(0deg);
        }
        to {
            -ms-transform: rotate(360deg);
            -moz-transform: rotate(360deg);
            -webkit-transform: rotate(360deg);
            -o-transform: rotate(360deg);
            transform: rotate(360deg);
        }
    }div.svelte-16h2a8k {width:80px;height:80px;
        -webkit-animation: svelte-16h2a8k-rotating 2s ease-in-out infinite;
        -moz-animation: svelte-16h2a8k-rotating 2s ease-in-out infinite;
        -ms-animation: svelte-16h2a8k-rotating 2s ease-in-out infinite;
        -o-animation: svelte-16h2a8k-rotating 2s ease-in-out infinite;
        animation: svelte-16h2a8k-rotating 2s ease-in-out infinite;}`
};
function os(e) {
  D(e, is);
  var t = rs(), n = k(t);
  Mt(n, () => ns), _(e, t);
}
var as = /* @__PURE__ */ I('<div class="rcb-modal-loader svelte-l57ojl"><!></div>');
const ss = {
  hash: "svelte-l57ojl",
  code: ".rcb-modal-loader.svelte-l57ojl {width:100%;flex-grow:1;min-height:100%;display:flex;justify-content:center;align-items:center;}"
};
function In(e) {
  D(e, ss);
  var t = as(), n = k(t);
  os(n), _(e, t);
}
var ne = /* @__PURE__ */ ((e) => (e[e.UnknownError = 0] = "UnknownError", e[e.UserCancelledError = 1] = "UserCancelledError", e[e.StoreProblemError = 2] = "StoreProblemError", e[e.PurchaseNotAllowedError = 3] = "PurchaseNotAllowedError", e[e.PurchaseInvalidError = 4] = "PurchaseInvalidError", e[e.ProductNotAvailableForPurchaseError = 5] = "ProductNotAvailableForPurchaseError", e[e.ProductAlreadyPurchasedError = 6] = "ProductAlreadyPurchasedError", e[e.ReceiptAlreadyInUseError = 7] = "ReceiptAlreadyInUseError", e[e.InvalidReceiptError = 8] = "InvalidReceiptError", e[e.MissingReceiptFileError = 9] = "MissingReceiptFileError", e[e.NetworkError = 10] = "NetworkError", e[e.InvalidCredentialsError = 11] = "InvalidCredentialsError", e[e.UnexpectedBackendResponseError = 12] = "UnexpectedBackendResponseError", e[e.InvalidAppUserIdError = 14] = "InvalidAppUserIdError", e[e.OperationAlreadyInProgressError = 15] = "OperationAlreadyInProgressError", e[e.UnknownBackendError = 16] = "UnknownBackendError", e[e.InvalidAppleSubscriptionKeyError = 17] = "InvalidAppleSubscriptionKeyError", e[e.IneligibleError = 18] = "IneligibleError", e[e.InsufficientPermissionsError = 19] = "InsufficientPermissionsError", e[e.PaymentPendingError = 20] = "PaymentPendingError", e[e.InvalidSubscriberAttributesError = 21] = "InvalidSubscriberAttributesError", e[e.LogOutWithAnonymousUserError = 22] = "LogOutWithAnonymousUserError", e[e.ConfigurationError = 23] = "ConfigurationError", e[e.UnsupportedError = 24] = "UnsupportedError", e[e.EmptySubscriberAttributesError = 25] = "EmptySubscriberAttributesError", e[e.CustomerInfoError = 28] = "CustomerInfoError", e[e.SignatureVerificationError = 36] = "SignatureVerificationError", e[e.InvalidEmailError = 38] = "InvalidEmailError", e))(ne || {});
class xt {
  static getPublicMessage(t) {
    switch (t) {
      case 0:
        return "Unknown error.";
      case 1:
        return "Purchase was cancelled.";
      case 2:
        return "There was a problem with the store.";
      case 3:
        return "The device or user is not allowed to make the purchase.";
      case 4:
        return "One or more of the arguments provided are invalid.";
      case 5:
        return "The product is not available for purchase.";
      case 6:
        return "This product is already active for the user.";
      case 7:
        return "There is already another active subscriber using the same receipt.";
      case 8:
        return "The receipt is not valid.";
      case 9:
        return "The receipt is missing.";
      case 10:
        return "Error performing request. Please check your network connection and try again.";
      case 11:
        return "There was a credentials issue. Check the underlying error for more details.";
      case 12:
        return "Received unexpected response from the backend.";
      case 14:
        return "The app user id is not valid.";
      case 15:
        return "The operation is already in progress.";
      case 16:
        return "There was an unknown backend error.";
      case 17:
        return "Apple Subscription Key is invalid or not present. In order to provide subscription offers, you must first generate a subscription key. Please see https://docs.revenuecat.com/docs/ios-subscription-offers for more info.";
      case 18:
        return "The User is ineligible for that action.";
      case 19:
        return "App does not have sufficient permissions to make purchases.";
      case 20:
        return "The payment is pending.";
      case 21:
        return "One or more of the attributes sent could not be saved.";
      case 22:
        return "Called logOut but the current user is anonymous.";
      case 23:
        return "There is an issue with your configuration. Check the underlying error for more details.";
      case 24:
        return "There was a problem with the operation. Looks like we doesn't support that yet. Check the underlying error for more details.";
      case 25:
        return "A request for subscriber attributes returned none.";
      case 28:
        return "There was a problem related to the customer info.";
      case 36:
        return "Request failed signature verification. Please see https://rev.cat/trusted-entitlements for more info.";
      case 38:
        return "Email is not valid. Please provide a valid email address.";
    }
  }
  static getErrorCodeForBackendErrorCode(t) {
    switch (t) {
      case 7101:
      case 7773:
        return 2;
      case 7102:
        return 7;
      case 7103:
        return 8;
      case 7107:
      case 7224:
      case 7225:
        return 11;
      case 7105:
      case 7106:
      case 7814:
        return 4;
      case 7772:
        return 6;
      case 7220:
        return 14;
      case 7229:
        return 2;
      case 7230:
      case 7e3:
        return 23;
      case 7231:
        return 2;
      case 7232:
        return 18;
      case 7263:
      case 7264:
        return 21;
      case 7104:
      case 7234:
      case 7226:
      case 7110:
        return 12;
      case 7662:
        return 24;
      case 7012:
      case 7834:
        return 38;
    }
  }
  static convertCodeToBackendErrorCode(t) {
    return t in xi ? t : null;
  }
  static convertPurchaseFlowErrorCodeToErrorCode(t) {
    switch (t) {
      case ie.ErrorSettingUpPurchase:
        return 2;
      case ie.ErrorChargingPayment:
        return 20;
      case ie.NetworkError:
        return 10;
      case ie.MissingEmailError:
        return 4;
      case ie.StripeError:
        return 2;
      case ie.UnknownError:
        return 0;
      case ie.AlreadyPurchasedError:
        return 6;
    }
  }
}
var xi = /* @__PURE__ */ ((e) => (e[e.BackendInvalidPlatform = 7e3] = "BackendInvalidPlatform", e[e.BackendInvalidEmail = 7012] = "BackendInvalidEmail", e[e.BackendStoreProblem = 7101] = "BackendStoreProblem", e[e.BackendCannotTransferPurchase = 7102] = "BackendCannotTransferPurchase", e[e.BackendInvalidReceiptToken = 7103] = "BackendInvalidReceiptToken", e[e.BackendInvalidAppStoreSharedSecret = 7104] = "BackendInvalidAppStoreSharedSecret", e[e.BackendInvalidPaymentModeOrIntroPriceNotProvided = 7105] = "BackendInvalidPaymentModeOrIntroPriceNotProvided", e[e.BackendProductIdForGoogleReceiptNotProvided = 7106] = "BackendProductIdForGoogleReceiptNotProvided", e[e.BackendInvalidPlayStoreCredentials = 7107] = "BackendInvalidPlayStoreCredentials", e[e.BackendInternalServerError = 7110] = "BackendInternalServerError", e[e.BackendEmptyAppUserId = 7220] = "BackendEmptyAppUserId", e[e.BackendInvalidAuthToken = 7224] = "BackendInvalidAuthToken", e[e.BackendInvalidAPIKey = 7225] = "BackendInvalidAPIKey", e[e.BackendBadRequest = 7226] = "BackendBadRequest", e[e.BackendPlayStoreQuotaExceeded = 7229] = "BackendPlayStoreQuotaExceeded", e[e.BackendPlayStoreInvalidPackageName = 7230] = "BackendPlayStoreInvalidPackageName", e[e.BackendPlayStoreGenericError = 7231] = "BackendPlayStoreGenericError", e[e.BackendUserIneligibleForPromoOffer = 7232] = "BackendUserIneligibleForPromoOffer", e[e.BackendInvalidAppleSubscriptionKey = 7234] = "BackendInvalidAppleSubscriptionKey", e[e.BackendInvalidSubscriberAttributes = 7263] = "BackendInvalidSubscriberAttributes", e[e.BackendInvalidSubscriberAttributesBody = 7264] = "BackendInvalidSubscriberAttributesBody", e[e.BackendProductIDsMalformed = 7662] = "BackendProductIDsMalformed", e[e.BackendAlreadySubscribedError = 7772] = "BackendAlreadySubscribedError", e[e.BackendPaymentGatewayGenericError = 7773] = "BackendPaymentGatewayGenericError", e[e.BackendOfferNotFound = 7814] = "BackendOfferNotFound", e[e.BackendNoMXRecordsFound = 7834] = "BackendNoMXRecordsFound", e))(xi || {});
class Z extends Error {
  constructor(n, r, i, o) {
    super(r);
    R(this, "toString", () => `PurchasesError(code: ${ne[this.errorCode]}, message: ${this.message})`);
    this.errorCode = n, this.underlyingErrorMessage = i, this.extra = o;
  }
  /** @internal */
  static getForBackendError(n, r) {
    const i = xt.getErrorCodeForBackendErrorCode(n);
    return new Z(
      i,
      xt.getPublicMessage(i),
      r,
      { backendErrorCode: n }
    );
  }
  /** @internal */
  static getForPurchasesFlowError(n) {
    return new Z(
      xt.convertPurchaseFlowErrorCodeToErrorCode(
        n.errorCode
      ),
      n.message,
      n.underlyingErrorMessage
    );
  }
}
class ls extends Error {
  constructor() {
    super("Purchases must be configured before calling getInstance");
  }
}
var wt = /* @__PURE__ */ ((e) => (e.Started = "started", e.InProgress = "in_progress", e.Succeeded = "succeeded", e.Failed = "failed", e))(wt || {}), tt = /* @__PURE__ */ ((e) => (e[e.SetupIntentCreationFailed = 1] = "SetupIntentCreationFailed", e[e.PaymentMethodCreationFailed = 2] = "PaymentMethodCreationFailed", e[e.PaymentChargeFailed = 3] = "PaymentChargeFailed", e[e.SetupIntentCompletionFailed = 4] = "SetupIntentCompletionFailed", e[e.AlreadyPurchased = 5] = "AlreadyPurchased", e))(tt || {});
function cs(e) {
  var t;
  return e.operation.redemption_info ? {
    redeemUrl: ((t = e.operation.redemption_info) == null ? void 0 : t.redeem_url) ?? null
  } : null;
}
var ie = /* @__PURE__ */ ((e) => (e[e.ErrorSettingUpPurchase = 0] = "ErrorSettingUpPurchase", e[e.ErrorChargingPayment = 1] = "ErrorChargingPayment", e[e.UnknownError = 2] = "UnknownError", e[e.NetworkError = 3] = "NetworkError", e[e.StripeError = 4] = "StripeError", e[e.MissingEmailError = 5] = "MissingEmailError", e[e.AlreadyPurchasedError = 6] = "AlreadyPurchasedError", e))(ie || {});
class Q extends Error {
  constructor(t, n, r, i, o) {
    super(n), this.errorCode = t, this.underlyingErrorMessage = r, this.purchasesErrorCode = i, this.extra = o;
  }
  isRecoverable() {
    switch (this.errorCode) {
      case 3:
      case 5:
        return !0;
      case 0:
      case 1:
      case 6:
      case 4:
      case 2:
        return !1;
    }
  }
  getPublicErrorMessage(t) {
    var r;
    const n = ((r = this.extra) == null ? void 0 : r.backendErrorCode) ?? this.purchasesErrorCode ?? this.errorCode;
    switch (this.errorCode) {
      case 2:
        return `An unknown error occurred. Error code: ${n}.`;
      case 0:
        return `Purchase not started due to an error. Error code: ${n}.`;
      case 1:
        return "Payment failed.";
      case 3:
        return "Network error. Please check your internet connection.";
      case 4:
        return this.message;
      case 5:
        return "Email is required to complete the purchase.";
      case 6:
        return (t == null ? void 0 : t.productType) === Bt.Subscription ? "You are already subscribed to this product." : "You have already purchased this product.";
    }
  }
  static fromPurchasesError(t, n) {
    let r;
    return t.errorCode === ne.ProductAlreadyPurchasedError ? r = 6 : t.errorCode === ne.InvalidEmailError ? r = 5 : t.errorCode === ne.NetworkError ? r = 3 : r = n, new Q(
      r,
      t.message,
      t.underlyingErrorMessage,
      t.errorCode,
      t.extra
    );
  }
}
class us {
  constructor(t, n = 10) {
    R(this, "operationSessionId", null);
    R(this, "backend");
    R(this, "maxNumberAttempts");
    R(this, "waitMSBetweenAttempts", 1e3);
    this.backend = t, this.maxNumberAttempts = n;
  }
  async startPurchase(t, n, r, i, o) {
    try {
      const a = await this.backend.postPurchase(
        t,
        n,
        i,
        o,
        r
      );
      return this.operationSessionId = a.operation_session_id, a;
    } catch (a) {
      if (a instanceof Z)
        throw Q.fromPurchasesError(
          a,
          0
          /* ErrorSettingUpPurchase */
        );
      {
        const s = "Unknown error starting purchase: " + String(a);
        throw F.errorLog(s), new Q(
          2,
          s
        );
      }
    }
  }
  async pollCurrentPurchaseForCompletion() {
    const t = this.operationSessionId;
    if (!t)
      throw new Q(
        0,
        "No purchase in progress"
      );
    return new Promise(
      (n, r) => {
        const i = (o = 1) => {
          if (o > this.maxNumberAttempts) {
            this.clearPurchaseInProgress(), r(
              new Q(
                2,
                "Max attempts reached trying to get successful purchase status"
              )
            );
            return;
          }
          this.backend.getCheckoutStatus(t).then((a) => {
            switch (a.operation.status) {
              case wt.Started:
              case wt.InProgress:
                setTimeout(
                  () => i(o + 1),
                  this.waitMSBetweenAttempts
                );
                break;
              case wt.Succeeded:
                this.clearPurchaseInProgress(), n({
                  redemptionInfo: cs(a)
                });
                return;
              case wt.Failed:
                this.clearPurchaseInProgress(), this.handlePaymentError(
                  a.operation.error,
                  r
                );
            }
          }).catch((a) => {
            const s = Q.fromPurchasesError(
              a,
              3
              /* NetworkError */
            );
            r(s);
          });
        };
        i();
      }
    );
  }
  clearPurchaseInProgress() {
    this.operationSessionId = null;
  }
  handlePaymentError(t, n) {
    if (t == null) {
      n(
        new Q(
          2,
          "Got an error status but error field is empty."
        )
      );
      return;
    }
    switch (t.code) {
      case tt.SetupIntentCreationFailed:
        n(
          new Q(
            0,
            "Setup intent creation failed"
          )
        );
        return;
      case tt.PaymentMethodCreationFailed:
        n(
          new Q(
            0,
            "Payment method creation failed"
          )
        );
        return;
      case tt.PaymentChargeFailed:
        n(
          new Q(
            1,
            "Payment charge failed"
          )
        );
        return;
      case tt.SetupIntentCompletionFailed:
        n(
          new Q(
            0,
            "Setup intent completion failed"
          )
        );
        return;
      case tt.AlreadyPurchased:
        n(
          new Q(
            6,
            "Purchased was already completed"
          )
        );
        return;
      default:
        n(
          new Q(
            2,
            "Unknown error code received"
          )
        );
        return;
    }
  }
}
const fs = `<svg width="96" height="96" viewBox="0 0 124 124" fill="currentColor" xmlns="http://www.w3.org/2000/svg">
    <circle cx="62" cy="62" r="62" fill="currentColor"/>
    <path d="M87.6668 41.504L82.4968 36.334L62.0002 56.8307L41.5035 36.334L36.3335 41.504L56.8302 62.0006L36.3335 82.4973L41.5035 87.6673L62.0002 67.1706L82.4968 87.6673L87.6668 82.4973L67.1702 62.0006L87.6668 41.504Z"
          fill="white"/>
</svg>
`;
var ds = /* @__PURE__ */ I('<div style="color:var(--rc-color-error);" class="rcb-ui-asset-icon svelte-7wiquz"><!></div>');
const vs = {
  hash: "svelte-7wiquz",
  code: ".rcb-ui-asset-icon.svelte-7wiquz {width:96px;height:96px;margin:0 auto;}"
};
function gs(e) {
  D(e, vs);
  var t = ds(), n = k(t);
  Mt(n, () => fs), _(e, t);
}
var hs = /* @__PURE__ */ I("<button><!></button>");
const ps = {
  hash: "svelte-2xwwo",
  code: `button.svelte-2xwwo {border:none;border-radius:var(--rc-shape-input-button-border-radius);font-size:16px;cursor:pointer;height:56px;color:var(--rc-color-grey-text-dark);background-color:var(--rc-color-grey-ui-dark);display:flex;align-items:center;justify-content:center;-webkit-tap-highlight-color:rgba(0, 0, 0, 0);-webkit-tap-highlight-color:transparent;}button.svelte-2xwwo:focus,
  button.svelte-2xwwo:active {outline:2px solid var(--rc-color-focus);}button.intent-primary.svelte-2xwwo {background-color:var(--rc-color-primary);color:var(--rc-color-primary-text);font-size:16px;}button.svelte-2xwwo:disabled {color:var(--rc-color-grey-text-light);background-color:var(--rc-color-grey-ui-dark);}button.intent-primary.svelte-2xwwo:disabled {color:var(--rc-color-grey-text-light);background-color:var(--rc-color-grey-ui-dark);}`
};
function Kn(e, t) {
  D(e, ps);
  let n = y(t, "intent", 8, "primary"), r = y(t, "disabled", 8, !1), i = y(t, "testId", 8, void 0), o = y(t, "type", 8, void 0);
  var a = hs(), s = k(a);
  W(s, t, "default", {}), B(() => {
    Yn(a, `${`intent-${n()}` ?? ""} svelte-2xwwo`), a.disabled = r(), G(a, "data-testid", i()), G(a, "type", o());
  }), on("click", a, function(c) {
    yi.call(this, t, c);
  }), _(e, a);
}
var _s = /* @__PURE__ */ I('<footer class="rcb-modal-footer svelte-1f9z0o8"><!></footer>');
const bs = {
  hash: "svelte-1f9z0o8",
  code: "footer.svelte-1f9z0o8 {display:flex;flex-direction:column;}"
};
function Qn(e, t) {
  D(e, bs);
  var n = _s(), r = k(n);
  W(r, t, "default", {}), _(e, n);
}
var ms = /* @__PURE__ */ I('<div class="column svelte-6o8e90"><!></div>');
const ys = {
  hash: "svelte-6o8e90",
  code: '.column.svelte-6o8e90 {display:flex;flex-direction:column;justify-content:flex-start;gap:var(--gap, "8px");flex-grow:1;}'
};
function It(e, t) {
  D(e, ys);
  let n = y(t, "gutter", 8, "8px");
  var r = ms(), i = k(r);
  W(i, t, "default", {}), B(() => G(r, "style", `--gap:${n()};`)), _(e, r);
}
const ws = "data:image/svg+xml,%3csvg%20width='24'%20height='24'%20viewBox='0%200%2024%2024'%20fill='none'%20xmlns='http://www.w3.org/2000/svg'%3e%3cg%20id='mdi_close'%3e%3cpath%20id='Vector'%20d='M19%206.41L17.59%205L12%2010.59L6.41%205L5%206.41L10.59%2012L5%2017.59L6.41%2019L12%2013.41L17.59%2019L19%2017.59L13.41%2012L19%206.41Z'%20fill='%23767676'/%3e%3c/g%3e%3c/svg%3e";
var Es = /* @__PURE__ */ I(`<button class="close-button svelte-1hv9ta1"><img${Sa("src", ws)} alt="close"></button>`);
const xs = {
  hash: "svelte-1hv9ta1",
  code: ".close-button.svelte-1hv9ta1 {border:none;cursor:pointer;background-color:transparent;padding:0px;height:24px;border-radius:50%;}.close-button.svelte-1hv9ta1:focus {outline:2px solid var(--rc-color-focus);}"
};
function Wn(e, t) {
  D(e, xs);
  let n = y(t, "disabled", 8, !1);
  var r = Es();
  B(() => r.disabled = n()), on("click", r, function(i) {
    yi.call(this, t, i);
  }), _(e, r);
}
var Is = /* @__PURE__ */ I('<picture class="rcb-app-icon-picture-container svelte-44qpq9"><source type="image/webp"> <img class="rcb-app-icon svelte-44qpq9" alt="App icon"></picture>'), Ps = /* @__PURE__ */ I('<div class="rcb-app-icon loading svelte-44qpq9"></div>');
const ks = {
  hash: "svelte-44qpq9",
  code: ".rcb-app-icon.svelte-44qpq9 {width:40px;height:40px;border-radius:12px;box-shadow:0px 1px 10px 0px rgba(0, 0, 0, 0.1);margin-right:16px;}.rcb-app-icon-picture-container.svelte-44qpq9 {height:40px;}.rcb-app-icon.loading.svelte-44qpq9 {background-color:gray;}"
};
function Nr(e, t) {
  D(e, ks);
  let n = y(t, "src", 8, null), r = y(t, "srcWebp", 8, null);
  var i = te(), o = U(i);
  C(
    o,
    () => n() !== null,
    (a) => {
      var s = Is(), c = k(s), l = N(c, 2);
      B(() => {
        G(c, "srcset", r()), G(l, "src", n());
      }), _(a, s);
    },
    (a) => {
      var s = Ps();
      _(a, s);
    }
  ), _(e, i);
}
const Sr = (e) => `https://da08ctfrofx1b.cloudfront.net/${e}`;
var Ts = /* @__PURE__ */ I('<!> <span class="app-title svelte-10uf5fq"> </span>', 1), As = /* @__PURE__ */ I('<div class="rcb-header-layout__business-info svelte-10uf5fq"><!></div>');
const Os = {
  hash: "svelte-10uf5fq",
  code: ".app-title.svelte-10uf5fq {font-weight:500;margin:8px 0;font-size:16px;}.rcb-header-layout__business-info.svelte-10uf5fq {display:flex;align-items:center;}"
};
function Ii(e, t) {
  Y(t, !1), D(e, Os);
  let n = y(t, "brandingInfo", 8, null);
  ye();
  var r = As(), i = k(r);
  C(
    i,
    () => n() !== null,
    (o) => {
      var a = Ts(), s = U(a);
      C(s, () => n().app_icon_webp !== null && n().app_icon !== null, (d) => {
        var u = /* @__PURE__ */ ge(() => Sr(n().app_icon)), f = /* @__PURE__ */ ge(() => Sr(n().app_icon_webp));
        Nr(d, {
          get src() {
            return g(u);
          },
          get srcWebp() {
            return g(f);
          }
        });
      });
      var c = N(s, 2), l = k(c);
      B(() => re(l, n().seller_company_name)), _(o, a);
    },
    (o) => {
      Nr(o, {});
    }
  ), _(e, r), $();
}
var Ns = /* @__PURE__ */ I('<div class="rcb-post-purchase-header-layout svelte-17puakv"><!> <!></div>');
const Ss = {
  hash: "svelte-17puakv",
  code: ".rcb-post-purchase-header-layout.svelte-17puakv {display:flex;justify-content:space-between;align-items:center;width:100%;margin-top:-4px;}"
};
function Rs(e, t) {
  D(e, Ss);
  let n = y(t, "brandingInfo", 8, null), r = y(t, "onClose", 8);
  ft(e, {
    as: "header",
    children: (i, o) => {
      var a = Ns(), s = k(a);
      Ii(s, {
        get brandingInfo() {
          return n();
        }
      });
      var c = N(s, 2);
      Wn(c, {
        $$events: {
          click(...l) {
            var d;
            (d = r()) == null || d.apply(this, l);
          }
        }
      }), _(i, a);
    },
    $$slots: { default: !0 }
  });
}
var Ls = /* @__PURE__ */ I('<span class="title svelte-idou50"> </span> <span class="subtitle svelte-idou50"><!></span>', 1), Us = /* @__PURE__ */ I("<!> <!>", 1), Ds = /* @__PURE__ */ I('<div class="rcb-modal-message svelte-idou50"><!></div>'), Ms = /* @__PURE__ */ I("<!> <!> <!>", 1);
const Cs = {
  hash: "svelte-idou50",
  code: ".rcb-modal-message.svelte-idou50 {width:100%;min-height:160px;display:flex;justify-content:center;align-items:center;flex-direction:column;text-align:center;margin-bottom:16px;margin-top:16px;}.title.svelte-idou50 {font-size:24px;line-height:1.25em;}.subtitle.svelte-idou50 {font-size:16px;line-height:1.25em;overflow-wrap:anywhere;}"
};
function Pi(e, t) {
  D(e, Cs);
  let n = y(t, "brandingInfo", 8, null), r = y(t, "onContinue", 8), i = y(t, "title", 8), o = y(t, "type", 8), a = y(t, "closeButtonTitle", 8, "Go back to app");
  It(e, {
    gutter: "32px",
    children: (s, c) => {
      var l = Ms(), d = U(l);
      Rs(d, {
        get brandingInfo() {
          return n();
        },
        get onClose() {
          return r();
        }
      });
      var u = N(d, 2);
      ft(u, {
        children: (v, p) => {
          var b = Ds(), x = k(b);
          It(x, {
            gutter: "48px",
            children: (m, E) => {
              var P = Us(), h = U(P);
              W(h, t, "icon", {});
              var w = N(h, 2);
              It(w, {
                gutter: "16px",
                children: (M, j) => {
                  var T = Ls(), H = U(T), z = k(H), ce = N(H, 2), A = k(ce);
                  W(A, t, "default", {}), B(() => re(z, i())), _(M, T);
                },
                $$slots: { default: !0 }
              }), _(m, P);
            },
            $$slots: { default: !0 }
          }), B(() => G(b, "data-type", o())), _(v, b);
        },
        $$slots: { default: !0 }
      });
      var f = N(u, 2);
      Qn(f, {
        children: (v, p) => {
          Kn(v, {
            type: "submit",
            $$events: {
              click(...b) {
                var x;
                (x = r()) == null || x.apply(this, b);
              }
            },
            children: (b, x) => {
              var m = He();
              B(() => re(m, a())), _(b, m);
            },
            $$slots: { default: !0 }
          });
        },
        $$slots: { default: !0 }
      }), _(s, l);
    },
    $$slots: { default: !0 }
  });
}
var Bs = /* @__PURE__ */ I("<br>If this error persists, please contact <a> </a>.", 1), qs = /* @__PURE__ */ I(" <!>", 1);
function js(e, t) {
  Y(t, !1);
  let n = y(t, "brandingInfo", 8, null), r = y(t, "lastError", 8), i = y(t, "supportEmail", 8, null), o = y(t, "productDetails", 8, null), a = y(t, "onContinue", 8);
  an(() => {
    F.errorLog(`Displayed error: ${ie[r().errorCode]}. Message: ${r().message ?? "None"}. Underlying error: ${r().underlyingErrorMessage ?? "None"}`);
  });
  function s() {
    var l;
    switch (r().errorCode) {
      case ie.AlreadyPurchasedError:
        return ((l = o()) == null ? void 0 : l.productType) === Bt.Subscription ? "Already subscribed" : "Already purchased";
      default:
        return "Something went wrong";
    }
  }
  ye();
  var c = /* @__PURE__ */ ge(s);
  Pi(e, {
    get title() {
      return g(c);
    },
    get brandingInfo() {
      return n();
    },
    get onContinue() {
      return a();
    },
    type: "error",
    closeButtonTitle: "Try Again",
    children: (l, d) => {
      var u = qs(), f = U(u);
      B(() => re(f, `${r().getPublicErrorMessage(o()) ?? ""} `));
      var v = N(f);
      C(v, i, (p) => {
        var b = Bs(), x = N(U(b), 2), m = k(x);
        B(() => {
          G(x, "href", `mailto:${i() ?? ""}`), re(m, i());
        }), _(p, b);
      }), _(l, u);
    },
    $$slots: {
      default: !0,
      icon: (l, d) => {
        gs(l);
      }
    }
  }), $();
}
const Fs = `<svg width="96" height="96" viewBox="0 0 124 124" fill="currentColor" xmlns="http://www.w3.org/2000/svg">
    <circle cx="62" cy="62" r="62" fill="currentColor"/>
    <rect x="44.1116" y="56" width="27.5" height="11" transform="rotate(45 44.1116 56)" fill="white"/>
    <rect x="79.1133" y="44.334" width="11" height="44" transform="rotate(45 79.1133 44.334)" fill="white"/>
</svg>
`;
var Hs = /* @__PURE__ */ I('<div style="color:var(--rc-color-accent);" class="rcb-ui-asset-icon svelte-7wiquz"><!></div>');
const Vs = {
  hash: "svelte-7wiquz",
  code: ".rcb-ui-asset-icon.svelte-7wiquz {width:96px;height:96px;margin:0 auto;}"
};
function Gs(e) {
  D(e, Vs);
  var t = Hs(), n = k(t);
  Mt(n, () => Fs), _(e, t);
}
function zs(e, t) {
  var a;
  Y(t, !1);
  let n = y(t, "productDetails", 8, null), r = y(t, "brandingInfo", 8, null), i = y(t, "onContinue", 8);
  const o = ((a = n()) == null ? void 0 : a.productType) === Bt.Subscription;
  ye(), Pi(e, {
    type: "success",
    title: "Purchase successful",
    get brandingInfo() {
      return r();
    },
    get onContinue() {
      return i();
    },
    closeButtonTitle: "Close",
    children: (s, c) => {
      var l = te(), d = U(l);
      C(d, () => o, (u) => {
        var f = He("Your plan is now active.");
        _(u, f);
      }), _(s, l);
    },
    $$slots: {
      default: !0,
      icon: (s, c) => {
        Gs(s);
      }
    }
  }), $();
}
function Ys(e, t, n, r, i = {}) {
  const o = n.create(t, i);
  return o.mount(e), o.on("change", (a) => r("change", a)), o.on("ready", (a) => r("ready", a)), o.on("focus", (a) => r("focus", a)), o.on("blur", (a) => r("blur", a)), o.on("escape", (a) => r("escape", a)), o.on("click", (a) => r("click", a)), o.on("loaderror", (a) => r("loaderror", a)), o.on("loaderstart", (a) => r("loaderstart", a)), o.on("networkschange", (a) => r("networkschange", a)), o;
}
const $s = typeof window > "u";
function Ks(e) {
  if (!$s)
    return e.registerAppInfo({
      name: "svelte-stripe-js",
      url: "https://svelte-stripe-js.vercel.app"
    });
}
var Qs = /* @__PURE__ */ I("<div></div>");
function Ws(e, t) {
  Y(t, !1);
  let n, r = he();
  const i = xa(), { elements: o } = Zo("stripe");
  let a = y(t, "options", 8, void 0);
  an(() => (n = Ys(g(r), "payment", o, i, a()), () => n.destroy()));
  function s() {
    n.blur();
  }
  function c() {
    n.clear();
  }
  function l() {
    n.destroy();
  }
  function d() {
    n.focus();
  }
  ye();
  var u = Qs();
  return wa(u, (f) => O(r, f), () => g(r)), _(e, u), Ft(t, "blur", s), Ft(t, "clear", c), Ft(t, "destroy", l), Ft(t, "focus", d), $({ blur: s, clear: c, destroy: l, focus: d });
}
function Zs(e, t) {
  Y(t, !1);
  const n = he();
  let r = y(t, "stripe", 8), i = y(t, "mode", 8, void 0), o = y(t, "theme", 8, "stripe"), a = y(t, "variables", 24, () => ({})), s = y(t, "rules", 24, () => ({})), c = y(t, "labels", 8, "above"), l = y(t, "loader", 8, "auto"), d = y(t, "fonts", 24, () => []), u = y(t, "locale", 8, "auto"), f = y(t, "currency", 8, void 0), v = y(t, "amount", 8, void 0), p = y(t, "clientSecret", 8, void 0), b = y(t, "elements", 12, null);
  Ke(
    () => (J(o()), J(a()), J(s()), J(c())),
    () => {
      O(n, {
        theme: o(),
        variables: a(),
        rules: s(),
        labels: c()
      });
    }
  ), Ke(
    () => (J(r()), J(b()), J(i()), J(f()), J(v()), g(n), J(p()), J(d()), J(l()), J(u()), gr),
    () => {
      r() && !b() && (b(r().elements({
        mode: i(),
        currency: f(),
        amount: v(),
        appearance: g(n),
        clientSecret: p(),
        fonts: d(),
        loader: l(),
        locale: u()
      })), Ks(r()), gr("stripe", {
        stripe: r(),
        elements: b()
      }));
    }
  ), Ke(
    () => (J(b()), g(n), J(u())),
    () => {
      b() && b().update({
        appearance: g(n),
        locale: u()
      });
    }
  ), Hn(), ye();
  var x = te(), m = U(x);
  C(m, () => r() && b(), (E) => {
    var P = te(), h = U(P);
    W(h, t, "default", {}), _(E, P);
  }), _(e, x), $();
}
var ki = "https://js.stripe.com/v3", Xs = /^https:\/\/js\.stripe\.com\/v3\/?(\?.*)?$/, Rr = "loadStripe.setLoadParameters was called but an existing Stripe.js script already exists in the document; existing script parameters will be used", Js = function() {
  for (var t = document.querySelectorAll('script[src^="'.concat(ki, '"]')), n = 0; n < t.length; n++) {
    var r = t[n];
    if (Xs.test(r.src))
      return r;
  }
  return null;
}, Lr = function(t) {
  var n = "", r = document.createElement("script");
  r.src = "".concat(ki).concat(n);
  var i = document.head || document.body;
  if (!i)
    throw new Error("Expected document.body not to be null. Stripe.js requires a <body> element.");
  return i.appendChild(r), r;
}, el = function(t, n) {
  !t || !t._registerWrapper || t._registerWrapper({
    name: "stripe-js",
    version: "4.6.0",
    startTime: n
  });
}, bt = null, Vt = null, Gt = null, tl = function(t) {
  return function() {
    t(new Error("Failed to load Stripe.js"));
  };
}, nl = function(t, n) {
  return function() {
    window.Stripe ? t(window.Stripe) : n(new Error("Stripe.js not available"));
  };
}, rl = function(t) {
  return bt !== null ? bt : (bt = new Promise(function(n, r) {
    if (typeof window > "u" || typeof document > "u") {
      n(null);
      return;
    }
    if (window.Stripe && t && console.warn(Rr), window.Stripe) {
      n(window.Stripe);
      return;
    }
    try {
      var i = Js();
      if (i && t)
        console.warn(Rr);
      else if (!i)
        i = Lr(t);
      else if (i && Gt !== null && Vt !== null) {
        var o;
        i.removeEventListener("load", Gt), i.removeEventListener("error", Vt), (o = i.parentNode) === null || o === void 0 || o.removeChild(i), i = Lr(t);
      }
      Gt = nl(n, r), Vt = tl(r), i.addEventListener("load", Gt), i.addEventListener("error", Vt);
    } catch (a) {
      r(a);
      return;
    }
  }), bt.catch(function(n) {
    return bt = null, Promise.reject(n);
  }));
}, il = function(t, n, r) {
  if (t === null)
    return null;
  var i = t.apply(void 0, n);
  return el(i, r), i;
}, mt, Ti = !1, Ai = function() {
  return mt || (mt = rl(null).catch(function(t) {
    return mt = null, Promise.reject(t);
  }), mt);
};
Promise.resolve().then(function() {
  return Ai();
}).catch(function(e) {
  Ti || console.warn(e);
});
var ol = function() {
  for (var t = arguments.length, n = new Array(t), r = 0; r < t; r++)
    n[r] = arguments[r];
  Ti = !0;
  var i = Date.now();
  return Ai().then(function(o) {
    return il(o, n, i);
  });
}, al = /* @__PURE__ */ I('<div class="rcb-header-layout svelte-1xe711j"><!></div>');
const sl = {
  hash: "svelte-1xe711j",
  code: ".rcb-header-layout.svelte-1xe711j {display:flex;justify-content:space-between;align-items:center;width:100%;font-size:24px;margin:0;font-weight:500;}"
};
function Zn(e, t) {
  D(e, sl), ft(e, {
    as: "header",
    children: (n, r) => {
      var i = al(), o = k(i);
      W(o, t, "default", {}), _(n, i);
    },
    $$slots: { default: !0 }
  });
}
const ll = `<svg width="24" height="24" viewBox="0 0 24 24" fill="currentColor"
     xmlns="http://www.w3.org/2000/svg">
    <path d="M18 8H17V6C17 3.24 14.76 1 12 1C9.24 1 7 3.24 7 6V8H6C4.9 8 4 8.9 4 10V20C4 21.1 4.9 22 6 22H18C19.1 22 20 21.1 20 20V10C20 8.9 19.1 8 18 8ZM12 17C10.9 17 10 16.1 10 15C10 13.9 10.9 13 12 13C13.1 13 14 13.9 14 15C14 16.1 13.1 17 12 17ZM15.1 8H8.9V6C8.9 4.29 10.29 2.9 12 2.9C13.71 2.9 15.1 4.29 15.1 6V8Z"
          fill="currentColor"/>
</svg>
`;
var cl = /* @__PURE__ */ I('<div style="color:var(--rc-color-accent);" class="rcb-ui-asset-icon svelte-1z0jca3"><!></div>');
const ul = {
  hash: "svelte-1z0jca3",
  code: "div.svelte-1z0jca3 {height:24px;width:24px;}"
};
function fl(e) {
  D(e, ul);
  var t = cl(), n = k(t);
  Mt(n, () => ll), _(e, t);
}
var dl = /* @__PURE__ */ I('<div class="processing svelte-rjmul6"></div>');
const vl = {
  hash: "svelte-rjmul6",
  code: `.processing.svelte-rjmul6 {width:12px;aspect-ratio:1;border-radius:50%;
    animation: svelte-rjmul6-l5 1.5s infinite linear;}
  @keyframes svelte-rjmul6-l5 {
    0% {
      box-shadow:
        20px 0 #fff2,
        -20px 0 #fff2;
      background: #fff2;
    }
    25% {
      box-shadow:
        20px 0 #fff2,
        -20px 0 #ffff;
      background: #fff2;
    }
    50% {
      box-shadow:
        20px 0 #fff2,
        -20px 0 #fff2;
      background: #ffff;
    }
    75% {
      box-shadow:
        20px 0 #ffff,
        -20px 0 #fff2;
      background: #fff2;
    }
    100% {
      box-shadow:
        20px 0 #fff2,
        -20px 0 #fff2;
      background: #fff2;
    }
  }`
};
function Oi(e) {
  D(e, vl);
  var t = dl();
  _(e, t);
}
const gl = {
  error: "#f2545b",
  warning: "#f4e971",
  focus: "#000080",
  accent: "#767676",
  primary: "#000000",
  "primary-text": "#ffffff",
  white: "#ffffff",
  "grey-text-dark": "rgba(0,0,0,1)",
  "grey-text-light": "rgba(0,0,0,0.5)",
  "grey-ui-dark": "rgba(0,0,0,0.125)",
  "grey-ui-light": "rgba(0,0,0,0.005)",
  "input-background": "transparent",
  background: "white"
}, hl = {
  error: "#f2545b",
  warning: "#f4e971",
  focus: "#000080",
  accent: "#767676",
  primary: "#ffffff",
  "primary-text": "#000000",
  white: "#ffffff",
  "grey-text-dark": "rgba(255,255,255,1)",
  "grey-text-light": "rgba(255,255,255,0.5)",
  "grey-ui-dark": "rgba(255,255,255,0.125)",
  "grey-ui-light": "rgba(255,255,255,0.005)",
  "input-background": "transparent",
  background: "#000000"
}, Ni = {
  error: "color_error",
  focus: "color_accent",
  accent: "color_accent",
  primary: "color_buttons_primary"
}, pl = {
  ...Ni,
  background: "color_form_bg"
}, _l = {
  ...Ni,
  background: "color_product_info_bg"
}, Si = {
  "input-border-radius": "12px",
  "input-button-border-radius": "12px",
  "modal-border-radius": "16px"
}, bl = {
  "input-border-radius": "0px",
  "input-button-border-radius": "0px",
  "modal-border-radius": "0px"
}, ml = {
  "input-border-radius": "24px",
  "input-button-border-radius": "56px",
  "modal-border-radius": "16px"
}, Ur = Si, Dr = (e) => e.length == 7 ? {
  r: parseInt(e.slice(1, 3), 16),
  g: parseInt(e.slice(3, 5), 16),
  b: parseInt(e.slice(5, 7), 16)
} : e.length == 4 ? {
  r: parseInt(e[1], 16),
  g: parseInt(e[2], 16),
  b: parseInt(e[3], 16)
} : null, Ri = ({
  r: e,
  g: t,
  b: n,
  luminanceThreshold: r
}) => {
  const i = (a) => (a = a / 255, a <= 0.03928 ? a / 12.92 : Math.pow((a + 0.055) / 1.055, 2.4));
  return 0.2126 * i(e) + 0.7152 * i(t) + 0.0722 * i(n) > r;
}, Li = 0.37, yl = (e, t = Li) => {
  const n = Ri({ ...e, luminanceThreshold: t }) ? "0,0,0" : "255,255,255";
  return {
    "grey-text-dark": `rgba(${n},1.0)`,
    "grey-text-light": `rgba(${n},0.50)`,
    "grey-ui-dark": `rgba(${n},0.125)`,
    "grey-ui-light": `rgba(${n},0.05)`
  };
}, wl = (e, t, n, r = Li) => {
  const i = {
    "grey-text-dark": n["grey-text-dark"],
    "grey-text-light": n["grey-text-light"],
    "grey-ui-dark": n["grey-ui-dark"],
    "grey-ui-light": n["grey-ui-light"],
    "primary-text": n["primary-text"]
  };
  if (e != null && e.startsWith("#")) {
    const o = Dr(e);
    o !== null && Object.assign(i, yl(o));
  }
  if (t != null && t.startsWith("#")) {
    const o = Dr(t);
    o !== null && (i["primary-text"] = Ri({ ...o, luminanceThreshold: r }) ? "black" : "white");
  }
  return i;
}, El = (e, t) => e || t, xl = (e, t, n) => {
  const r = Object.entries(e).map(([i, o]) => [
    i,
    El(
      n ? n[o] : null,
      t[i]
    )
  ]);
  return Object.fromEntries(r);
}, Ui = (e, t, n) => {
  const r = xl(
    e,
    t,
    n
  );
  return n ? {
    ...t,
    ...r,
    ...wl(
      r.background,
      r.primary,
      t
    )
  } : { ...t };
}, Il = (e) => Ui(
  _l,
  hl,
  e
), Di = (e) => Ui(
  pl,
  gl,
  e
), Xn = (e) => {
  if (!e)
    return Ur;
  switch (e.shapes) {
    case "rounded":
      return Si;
    case "rectangle":
      return bl;
    case "pill":
      return ml;
    default:
      return Ur;
  }
}, Wt = (e = "", t) => t.map(([n, r]) => `--rc-${e}-${n}: ${r}`).join("; "), Mi = (e) => {
  const t = Wt(
    "color",
    Object.entries(Il(e))
  ), n = Wt(
    "shape",
    Object.entries(Xn(e))
  );
  return [t, n].join("; ");
}, Pl = (e) => {
  const t = Wt(
    "color",
    Object.entries(Di(e))
  ), n = Wt(
    "shape",
    Object.entries(Xn(e))
  );
  return [t, n].join("; ");
};
class Jn {
  constructor(t) {
    this.brandingAppearance = t;
  }
  get shape() {
    return Xn(this.brandingAppearance);
  }
  get formColors() {
    return Di(this.brandingAppearance);
  }
  get formStyleVars() {
    return Pl(this.brandingAppearance);
  }
  get productInfoStyleVars() {
    return Mi(this.brandingAppearance);
  }
}
var kl = /* @__PURE__ */ I('<div style="display: flex; align-items: center; justify-content: baseline;"><!> <div style="margin-left: 10px">Secure Checkout</div></div> <!>', 1), Tl = /* @__PURE__ */ I('<div class="rcb-stripe-elements-container svelte-8jsoa8"><!></div>'), Al = /* @__PURE__ */ I("<!> <!>", 1), Ol = /* @__PURE__ */ I("<!> <form><!></form>", 1), Nl = /* @__PURE__ */ I("<div><!></div>");
const Sl = {
  hash: "svelte-8jsoa8",
  code: `.rcb-stripe-elements-container.svelte-8jsoa8 {width:100%;

        /* The standard height of the payment form from Stripe */
        /* Added to avoid the card getting smaller while loading */min-height:320px;margin-top:32px;margin-bottom:24px;}`
};
function Rl(e, t) {
  var h;
  Y(t, !1), D(e, Sl);
  let n = y(t, "onClose", 8), r = y(t, "onContinue", 8), i = y(t, "onError", 8), o = y(t, "paymentInfoCollectionMetadata", 8), a = y(t, "processing", 12, !1), s = y(t, "productDetails", 8), c = y(t, "purchaseOptionToUse", 8), l = y(t, "brandingInfo", 8);
  const d = o().data.client_secret;
  let u = he(null), f = he(), v = he();
  an(async () => {
    const w = o().data.publishable_api_key, M = o().data.stripe_account_id;
    if (!w || !M)
      throw new Error("Stripe publishable key or account ID not found");
    O(u, await ol(w, { stripeAccount: M }));
  });
  const p = async () => {
    if (a() || !g(u) || !g(v)) return;
    a(!0);
    const w = await g(u).confirmSetup({
      elements: g(v),
      redirect: "if_required"
    });
    w.error ? (a(!1), i()(new Q(ie.StripeError, w.error.message))) : r()();
  }, b = new Jn((h = l()) == null ? void 0 : h.appearance);
  let x = b.shape, m = b.formColors;
  Ke(() => g(f), () => {
    g(f) && g(f)._elements.length > 0 && O(v, g(f));
  }), Hn(), ye();
  var E = Nl(), P = k(E);
  C(
    P,
    () => g(u) && d,
    (w) => {
      var M = Ol(), j = U(M);
      Zn(j, {
        children: (A, V) => {
          var X = kl(), Te = U(X), Ue = k(Te);
          fl(Ue);
          var Ae = N(Te, 2);
          Wn(Ae, {
            $$events: {
              click(...Oe) {
                var we;
                (we = n()) == null || we.apply(this, Oe);
              }
            }
          }), _(A, X);
        },
        $$slots: { default: !0 }
      });
      var T = N(j, 2), H = k(T), z = /* @__PURE__ */ ge(() => ({
        borderRadius: x["input-border-radius"],
        fontSizeBase: "16px",
        fontSizeSm: "16px",
        spacingGridRow: "16px",
        focusBoxShadow: "none",
        colorDanger: m.error,
        colorTextPlaceholder: m["grey-text-light"]
      })), ce = /* @__PURE__ */ ge(() => ({
        ".Input": {
          boxShadow: "none",
          border: `2px solid ${m["grey-ui-dark"]}`,
          backgroundColor: "transparent",
          color: m["grey-text-dark"]
        },
        ".Input:focus": {
          border: `2px solid ${m.focus}`,
          outline: "none"
        },
        ".Label": {
          marginBottom: "8px",
          fontWeight: "500",
          lineHeight: "22px",
          color: m["grey-text-dark"]
        },
        ".Input--invalid": { boxShadow: "none" },
        ".Tab": {
          boxShadow: "none",
          backgroundColor: "transparent"
        }
      }));
      Zs(H, {
        get stripe() {
          return g(u);
        },
        clientSecret: d,
        loader: "always",
        locale: "en",
        get elements() {
          return g(f);
        },
        set elements(A) {
          O(f, A);
        },
        theme: "stripe",
        get variables() {
          return g(z);
        },
        get rules() {
          return g(ce);
        },
        children: (A, V) => {
          var X = Al(), Te = U(X);
          ft(Te, {
            children: (Ae, Oe) => {
              var we = Tl(), jt = k(we);
              Ws(jt, {}), _(Ae, we);
            },
            $$slots: { default: !0 }
          });
          var Ue = N(Te, 2);
          Qn(Ue, {
            children: (Ae, Oe) => {
              It(Ae, {
                children: (we, jt) => {
                  Kn(we, {
                    get disabled() {
                      return a();
                    },
                    testId: "PayButton",
                    children: (cn, Ve) => {
                      var dt = te(), _e = U(dt);
                      C(
                        _e,
                        a,
                        (De) => {
                          Oi(De);
                        },
                        (De) => {
                          var Ee = te(), Ge = U(Ee);
                          C(
                            Ge,
                            () => {
                              var ae, se;
                              return (se = (ae = s().subscriptionOptions) == null ? void 0 : ae[c().id]) == null ? void 0 : se.trial;
                            },
                            (ae) => {
                              var se = He("Start Trial");
                              _(ae, se);
                            },
                            (ae) => {
                              var se = He("Pay");
                              _(ae, se);
                            },
                            !0
                          ), _(De, Ee);
                        }
                      ), _(cn, dt);
                    },
                    $$slots: { default: !0 }
                  });
                },
                $$slots: { default: !0 }
              });
            },
            $$slots: { default: !0 }
          }), _(A, X);
        },
        $$slots: { default: !0 },
        $$legacy: !0
      }), on("submit", T, mi(p)), _(w, M);
    },
    (w) => {
      In(w);
    }
  ), _(e, E), $();
}
function Ll(e) {
  return e.trim() === "" ? "You need to provide your email address to continue." : e.match(/^[^@]+@[^@]+\.[^@]+$/) ? null : "Email is not valid. Please provide a valid email address.";
}
function Ul(e) {
  if (e && !e.match(/^[A-Z]{3}$/)) {
    const t = `Currency code ${e} is not valid. Please provide a valid ISO 4217 currency code.`;
    throw F.errorLog(t), new Z(ne.ConfigurationError, t);
  }
}
var Dl = /* @__PURE__ */ I("<span>Billing email address</span> <!>", 1), Ml = /* @__PURE__ */ I('<div class="form-error svelte-yf01ty"> </div>'), Cl = /* @__PURE__ */ I('<div class="form-container svelte-yf01ty"><div class="form-label svelte-yf01ty"><label for="email">Email</label></div> <div><input name="email" placeholder="john@appleseed.com" autocapitalize="off" class="svelte-yf01ty"></div> <!></div>'), Bl = /* @__PURE__ */ I('<div class="container svelte-yf01ty"><!> <form class="svelte-yf01ty"><!> <!></form></div>');
const ql = {
  hash: "svelte-yf01ty",
  code: ".container.svelte-yf01ty {display:flex;flex-direction:column;flex-grow:1;}form.svelte-yf01ty {display:flex;flex-direction:column;min-height:100%;flex-grow:1;}.form-container.svelte-yf01ty {display:flex;flex-direction:column;width:100%;margin-top:32px;margin-bottom:16px;}.form-label.svelte-yf01ty {margin-top:8px;margin-bottom:8px;display:block;font-weight:500;line-height:22px;}.form-input.error.svelte-yf01ty input:where(.svelte-yf01ty) {border-color:var(--rc-color-error);}.form-error.svelte-yf01ty {margin-top:4px;font-size:16px;line-height:20px;min-height:40px;color:var(--rc-color-error);}input.svelte-yf01ty {width:100%;box-sizing:border-box;padding:8px;border:2px solid var(--rc-color-grey-ui-dark);border-radius:var(--rc-shape-input-border-radius);font-size:16px;height:48px;padding:6px 14px;background:var(--rc-color-input-background);color:inherit;}input.svelte-yf01ty:focus {outline:none;border:2px solid var(--rc-color-focus);}"
};
function jl(e, t) {
  Y(t, !1), D(e, ql);
  const n = he(), r = he(), i = he();
  let o = y(t, "onContinue", 8), a = y(t, "onClose", 8), s = y(t, "processing", 8), c = y(t, "lastError", 8);
  const l = async () => {
    const b = Ll(g(n));
    b ? O(r, b) : o()({ email: g(n) });
  };
  Ia(async () => {
    var b;
    O(r, ((b = c()) == null ? void 0 : b.message) ?? "");
  }), Ke(() => {
  }, () => {
    O(n, "");
  }), Ke(() => {
  }, () => {
    O(r, "");
  }), Ke(() => g(r), () => {
    O(i, g(r) ? "error" : "");
  }), Hn(), ye();
  var d = Bl(), u = k(d);
  Zn(u, {
    children: (b, x) => {
      var m = Dl(), E = N(U(m), 2);
      Wn(E, {
        $$events: {
          click(...P) {
            var h;
            (h = a()) == null || h.apply(this, P);
          }
        }
      }), _(b, m);
    },
    $$slots: { default: !0 }
  });
  var f = N(u, 2), v = k(f);
  ft(v, {
    children: (b, x) => {
      var m = Cl(), E = N(k(m), 2), P = k(E), h = N(E, 2);
      C(h, () => g(r), (w) => {
        var M = Ml(), j = k(M);
        B(() => re(j, g(r))), _(w, M);
      }), B(() => Yn(E, `form-input ${g(i) ?? ""} svelte-yf01ty`)), ya(P, () => g(n), (w) => O(n, w)), _(b, m);
    },
    $$slots: { default: !0 }
  });
  var p = N(v, 2);
  Qn(p, {
    children: (b, x) => {
      It(b, {
        children: (m, E) => {
          Kn(m, {
            get disabled() {
              return s();
            },
            type: "submit",
            children: (P, h) => {
              var w = te(), M = U(w);
              C(
                M,
                s,
                (j) => {
                  Oi(j);
                },
                (j) => {
                  var T = He("Continue");
                  _(j, T);
                }
              ), _(P, w);
            },
            $$slots: { default: !0 }
          });
        },
        $$slots: { default: !0 }
      });
    },
    $$slots: { default: !0 }
  }), on("submit", f, mi(l)), _(e, d), $();
}
var Fl = /* @__PURE__ */ I('<div class="rcb-modal-backdrop svelte-17tslso"><!></div>');
const Hl = {
  hash: "svelte-17tslso",
  code: ".rcb-modal-backdrop.svelte-17tslso {position:fixed;top:0;left:0;right:0;min-width:100%;width:100vw;width:100dvw;min-height:100%;height:100vh;height:100dvh;background-color:rgba(40, 40, 40, 0.75);display:flex;flex-direction:column;justify-content:space-around;z-index:1000001;}"
};
function Vl(e, t) {
  D(e, Hl);
  var n = Fl(), r = k(n);
  W(r, t, "default", {}), _(e, n);
}
var Gl = /* @__PURE__ */ I('<div class="rcb-modal-container svelte-1fufj5b"><main class="rcb-modal-main svelte-1fufj5b"><!></main></div>');
const zl = {
  hash: "svelte-1fufj5b",
  code: `.rcb-modal-container.svelte-1fufj5b {width:100%;}.rcb-modal-main.svelte-1fufj5b {box-sizing:border-box;border-radius:var(--rc-shape-modal-border-radius);background-color:var(--rc-color-background);color:var(--rc-color-grey-text-dark);padding:40px;flex-direction:column;display:flex;}

    @media screen and (max-width: 960px) {.rcb-modal-container.svelte-1fufj5b {width:calc(100% - 32px);min-width:300px;max-width:640px;margin:auto;}.rcb-modal-main.svelte-1fufj5b {min-width:300px;max-width:640px;padding:32px;}
    }`
};
function Ci(e, t) {
  D(e, zl);
  let n = y(t, "style", 8, "");
  var r = Gl(), i = k(r), o = k(i);
  W(o, t, "default", {}), B(() => G(i, "style", n())), _(e, r);
}
function Yl(e, t) {
  let n = y(t, "condition", 8, !1);
  var r = te(), i = U(r);
  C(
    i,
    n,
    (o) => {
      Vl(o, {
        children: (a, s) => {
          var c = te(), l = U(c);
          W(l, t, "default", {}), _(a, c);
        },
        $$slots: { default: !0 }
      });
    },
    (o) => {
      var a = te(), s = U(a);
      W(s, t, "default", {}), _(o, a);
    }
  ), _(e, r);
}
const $l = `<svg width="24" height="24" viewBox="0 0 24 24" fill="currentColor"
     xmlns="http://www.w3.org/2000/svg">
    <g opacity="0.5" clip-path="url(#clip0_344_3390)">
        <path d="M7 18C5.9 18 5.01 18.9 5.01 20C5.01 21.1 5.9 22 7 22C8.1 22 9 21.1 9 20C9 18.9 8.1 18 7 18ZM1 2V4H3L6.6 11.59L5.25 14.04C5.09 14.32 5 14.65 5 15C5 16.1 5.9 17 7 17H19V15H7.42C7.28 15 7.17 14.89 7.17 14.75L7.2 14.63L8.1 13H15.55C16.3 13 16.96 12.59 17.3 11.97L20.88 5.48C20.96 5.34 21 5.17 21 5C21 4.45 20.55 4 20 4H5.21L4.27 2H1ZM17 18C15.9 18 15.01 18.9 15.01 20C15.01 21.1 15.9 22 17 22C18.1 22 19 21.1 19 20C19 18.9 18.1 18 17 18Z"
              fill="currentColor"/>
    </g>
    <defs>
        <clipPath id="clip0_344_3390">
            <rect width="24" height="24" fill="currentColor"/>
        </clipPath>
    </defs>
</svg>
`;
var Kl = /* @__PURE__ */ I('<div style="color:var(--rc-color-gray-text-dark);"><!></div>');
function Ql(e) {
  var t = Kl(), n = k(t);
  Mt(n, () => $l), _(e, t);
}
var Wl = /* @__PURE__ */ I('<div class="svelte-1ny4y2s">Sandbox</div>');
const Zl = {
  hash: "svelte-1ny4y2s",
  code: "div.svelte-1ny4y2s {background-color:var(--rc-color-warning);color:rgba(0,0,0,1);font-size:10px;font-weight:bold;text-transform:uppercase;padding:4px 10px;line-height:10px;border-radius:9px;margin:0px;margin-left:16px;}"
};
function Xl(e) {
  D(e, Zl);
  var t = Wl();
  _(e, t);
}
var Jl = /* @__PURE__ */ I('<div class="rcb-ui-layout svelte-wuqzav"><!></div>');
const ec = {
  hash: "svelte-wuqzav",
  code: `.rcb-ui-layout.svelte-wuqzav {width:100vw;margin-right:auto;display:flex;justify-content:center;align-items:flex-start;position:relative;padding:0px 80px;box-sizing:border-box;}

    @media screen and (max-width: 960px) {.rcb-ui-layout.svelte-wuqzav {flex-direction:column;align-items:center;justify-content:center;height:100%;min-width:100%;padding:16px 0;}
    }

    @media screen and (max-width: 960px) and (max-height: 960px) {.rcb-ui-layout.svelte-wuqzav {overflow-y:scroll;justify-content:flex-start;}
    }`
};
function tc(e, t) {
  D(e, ec);
  let n = y(t, "style", 8, "");
  var r = Jl(), i = k(r);
  W(i, t, "default", {}), B(() => G(r, "style", n())), _(e, r);
}
var nc = /* @__PURE__ */ I('<div class="rcb-ui-container svelte-1sfb5if"><!></div>');
const rc = {
  hash: "svelte-1sfb5if",
  code: `.rcb-ui-container.svelte-1sfb5if {color-scheme:none;font-size:16px;line-height:1.5em;font-weight:400;font-family:-apple-system, "system-ui", "Segoe UI", Roboto, Oxygen, Ubuntu,
        Cantarell, "Open Sans", "Helvetica Neue", sans-serif;}`
};
function ic(e, t) {
  D(e, rc);
  let n = y(t, "style", 8, "");
  var r = nc(), i = k(r);
  W(i, t, "default", {}), B(() => G(r, "style", n())), _(e, r);
}
var oc = /* @__PURE__ */ I("<!> <!>", 1), ac = /* @__PURE__ */ I('<div class="rcb-ui-aside svelte-17dr902"><!></div>');
const sc = {
  hash: "svelte-17dr902",
  code: `.rcb-ui-aside.svelte-17dr902 {margin-right:16px;flex:320px 1 0;max-width:480px;}

    @media screen and (max-width: 960px) {.rcb-ui-aside.svelte-17dr902 {margin-right:0;margin-bottom:16px;min-width:100%;display:flex;flex-direction:column;align-items:center;flex:none;max-width:none;}
    }`
};
function lc(e, t) {
  Y(t, !1), D(e, sc);
  let n = y(t, "brandingAppearance", 8, void 0), r = new Jn(n()).productInfoStyleVars;
  ye();
  var i = ac();
  G(i, "style", r);
  var o = k(i);
  Ci(o, {
    children: (a, s) => {
      var c = oc(), l = U(c);
      W(l, t, "header", {});
      var d = N(l, 2);
      W(d, t, "default", {}), _(a, c);
    },
    $$slots: { default: !0 }
  }), _(e, i), $();
}
var cc = /* @__PURE__ */ I("<!> <!>", 1), uc = /* @__PURE__ */ I('<div class="rcb-ui-main svelte-vpgys"><!></div>');
const fc = {
  hash: "svelte-vpgys",
  code: `.rcb-ui-main.svelte-vpgys {flex:480px 1 0;max-width:640px;}

    @media screen and (max-width: 960px) {.rcb-ui-main.svelte-vpgys {flex:none;max-width:none;min-width:100%;}
    }`
};
function dc(e, t) {
  Y(t, !1), D(e, fc);
  let n = y(t, "brandingAppearance", 8, void 0), r = new Jn(n()).formStyleVars;
  ye();
  var i = uc();
  G(i, "style", r);
  var o = k(i);
  Ci(o, {
    style: "min-height: 360px;",
    children: (a, s) => {
      var c = cc(), l = U(c);
      W(l, t, "header", {});
      var d = N(l, 2);
      W(d, t, "default", {}), _(a, c);
    },
    $$slots: { default: !0 }
  }), _(e, i), $();
}
var vc = /* @__PURE__ */ I("<!> <!>", 1), gc = /* @__PURE__ */ I("<!> <!> <!> <!> <!> <!> <!>", 1), hc = /* @__PURE__ */ I("<!> <!>", 1);
function pc(e, t) {
  Y(t, !1);
  let n = y(t, "asModal", 8, !0), r = y(t, "customerEmail", 12), i = y(t, "appUserId", 8), o = y(t, "rcPackage", 8), a = y(t, "purchaseOption", 8), s = y(t, "brandingInfo", 8), c = y(t, "onFinished", 8), l = y(t, "onError", 8), d = y(t, "onClose", 8), u = y(t, "purchases", 8), f = y(t, "purchaseOperationHelper", 8), v = he(""), p = he(null), b = he(null), x = he(null);
  const m = o().rcBillingProduct.identifier ?? null, E = o().rcBillingProduct.defaultPurchaseOption, P = a() ? a() : E;
  let h = he("present-offer"), w = null;
  const M = [
    "present-offer",
    "needs-auth-info",
    "processing-auth-info",
    "needs-payment-info",
    "polling-purchase-status",
    "loading"
  ];
  an(async () => {
    var A;
    if (O(p, o().rcBillingProduct), O(v, Mi((A = s()) == null ? void 0 : A.appearance)), g(h) === "present-offer") {
      r() ? T() : O(h, "needs-auth-info");
      return;
    }
  });
  const j = () => {
    d()();
  }, T = () => {
    if (m === null) {
      z(new Q(ie.ErrorSettingUpPurchase, "Product ID was not set before purchase."));
      return;
    } else g(h) === "present-offer" && O(h, "loading");
    if (!r()) {
      z(new Q(ie.MissingEmailError));
      return;
    }
    f().startPurchase(i(), m, P, r(), o().rcBillingProduct.presentedOfferingContext).then((A) => {
      if (A.next_action === "collect_payment_info") {
        O(x, null), O(h, "needs-payment-info"), O(b, A);
        return;
      }
      if (A.next_action === "completed") {
        O(x, null), O(h, "success");
        return;
      }
    }).catch((A) => {
      z(A);
    });
  }, H = (A) => {
    if (g(h) === "needs-auth-info") {
      A && (r(A.email), O(h, "processing-auth-info")), T();
      return;
    }
    if (g(h) === "needs-payment-info") {
      O(h, "polling-purchase-status"), f().pollCurrentPurchaseForCompletion().then((V) => {
        O(h, "success"), w = V.redemptionInfo;
      }).catch((V) => {
        z(V);
      });
      return;
    }
    if (g(h) === "success" || g(h) === "error") {
      c()(w);
      return;
    }
    O(h, "success");
  }, z = (A) => {
    if (g(h) === "processing-auth-info" && A.isRecoverable()) {
      O(x, A), O(h, "needs-auth-info");
      return;
    }
    O(x, A), O(h, "error");
  }, ce = () => {
    l()(g(x) ?? new Q(ie.UnknownError, "Unknown error without state set."));
  };
  ye(), ic(e, {
    children: (A, V) => {
      Yl(A, {
        get condition() {
          return n();
        },
        children: (X, Te) => {
          tc(X, {
            get style() {
              return g(v);
            },
            children: (Ue, Ae) => {
              var Oe = hc(), we = U(Oe);
              C(we, () => M.includes(g(h)), (Ve) => {
                var dt = /* @__PURE__ */ ge(() => {
                  var _e;
                  return (_e = s()) == null ? void 0 : _e.appearance;
                });
                lc(Ve, {
                  get brandingAppearance() {
                    return g(dt);
                  },
                  children: (_e, De) => {
                    var Ee = te(), Ge = U(Ee);
                    C(Ge, () => g(p) && P, (ae) => {
                      var se = /* @__PURE__ */ ge(() => {
                        var ze;
                        return (ze = s()) == null ? void 0 : ze.appearance;
                      });
                      Or(ae, {
                        get productDetails() {
                          return g(p);
                        },
                        get brandingAppearance() {
                          return g(se);
                        },
                        purchaseOption: P
                      });
                    }), _(_e, Ee);
                  },
                  $$slots: {
                    default: !0,
                    header: (_e, De) => {
                      Zn(_e, {
                        slot: "header",
                        children: (Ee, Ge) => {
                          var ae = vc(), se = U(ae);
                          Ii(se, {
                            get brandingInfo() {
                              return s();
                            }
                          });
                          var ze = N(se, 2);
                          C(
                            ze,
                            () => u().isSandbox(),
                            (vt) => {
                              Xl(vt);
                            },
                            (vt) => {
                              Ql(vt);
                            }
                          ), _(Ee, ae);
                        },
                        $$slots: { default: !0 }
                      });
                    }
                  }
                });
              });
              var jt = N(we, 2), cn = /* @__PURE__ */ ge(() => {
                var Ve;
                return (Ve = s()) == null ? void 0 : Ve.appearance;
              });
              dc(jt, {
                get brandingAppearance() {
                  return g(cn);
                },
                children: (Ve, dt) => {
                  var _e = gc(), De = U(_e);
                  C(De, () => g(h) === "present-offer" && g(p) && P, (ue) => {
                    Or(ue, {
                      get productDetails() {
                        return g(p);
                      },
                      purchaseOption: P
                    });
                  });
                  var Ee = N(De, 2);
                  C(Ee, () => g(h) === "present-offer" && !g(p), (ue) => {
                    In(ue);
                  });
                  var Ge = N(Ee, 2);
                  C(Ge, () => g(h) === "needs-auth-info" || g(h) === "processing-auth-info", (ue) => {
                    var gt = /* @__PURE__ */ ge(() => g(h) === "processing-auth-info");
                    jl(ue, {
                      onContinue: H,
                      onClose: j,
                      get processing() {
                        return g(gt);
                      },
                      get lastError() {
                        return g(x);
                      }
                    });
                  });
                  var ae = N(Ge, 2);
                  C(ae, () => g(b) && (g(h) === "needs-payment-info" || g(h) === "polling-purchase-status") && g(p) && P, (ue) => {
                    var gt = /* @__PURE__ */ ge(() => g(h) === "polling-purchase-status");
                    Rl(ue, {
                      get paymentInfoCollectionMetadata() {
                        return g(b);
                      },
                      onContinue: H,
                      onClose: j,
                      onError: z,
                      get processing() {
                        return g(gt);
                      },
                      get productDetails() {
                        return g(p);
                      },
                      purchaseOptionToUse: P,
                      get brandingInfo() {
                        return s();
                      }
                    });
                  });
                  var se = N(ae, 2);
                  C(se, () => g(h) === "loading", (ue) => {
                    In(ue);
                  });
                  var ze = N(se, 2);
                  C(ze, () => g(h) === "error", (ue) => {
                    var gt = /* @__PURE__ */ ge(() => g(x) ?? new Q(ie.UnknownError, "Unknown error without state set.")), Zi = /* @__PURE__ */ ge(() => {
                      var nr;
                      return (nr = s()) == null ? void 0 : nr.seller_company_support_email;
                    });
                    js(ue, {
                      get brandingInfo() {
                        return s();
                      },
                      get lastError() {
                        return g(gt);
                      },
                      get supportEmail() {
                        return g(Zi);
                      },
                      get productDetails() {
                        return g(p);
                      },
                      onContinue: ce
                    });
                  });
                  var vt = N(ze, 2);
                  C(vt, () => g(h) === "success", (ue) => {
                    zs(ue, {
                      get productDetails() {
                        return g(p);
                      },
                      get brandingInfo() {
                        return s();
                      },
                      onContinue: H
                    });
                  }), _(Ve, _e);
                },
                $$slots: { default: !0 }
              }), _(Ue, Oe);
            },
            $$slots: { default: !0 }
          });
        },
        $$slots: { default: !0 }
      });
    },
    $$slots: { default: !0 }
  }), $();
}
function _c(e) {
  return e.expires_date == null ? !0 : new Date(e.expires_date) > /* @__PURE__ */ new Date();
}
function bc(e, t, n, r) {
  const i = t.product_identifier;
  if (i in n)
    return mc(
      e,
      t,
      n[i]
    );
  if (i in r)
    return yc(
      e,
      t,
      r[i]
    );
  throw new Z(
    ne.CustomerInfoError,
    "Could not find entitlement product in subscriptions or non-subscriptions."
  );
}
function mc(e, t, n) {
  return {
    identifier: e,
    isActive: _c(t),
    willRenew: xc(t, n),
    store: (n == null ? void 0 : n.store) ?? "unknown",
    latestPurchaseDate: new Date(t.purchase_date),
    originalPurchaseDate: new Date(t.purchase_date),
    expirationDate: zt(t.expires_date),
    productIdentifier: t.product_identifier,
    unsubscribeDetectedAt: zt(
      n == null ? void 0 : n.unsubscribe_detected_at
    ),
    billingIssueDetectedAt: zt(
      n == null ? void 0 : n.billing_issues_detected_at
    ),
    isSandbox: (n == null ? void 0 : n.is_sandbox) ?? !1,
    periodType: (n == null ? void 0 : n.period_type) ?? "normal"
  };
}
function yc(e, t, n) {
  return {
    identifier: e,
    isActive: !0,
    willRenew: !1,
    store: n.store,
    latestPurchaseDate: new Date(t.purchase_date),
    originalPurchaseDate: new Date(
      n.original_purchase_date
    ),
    expirationDate: null,
    productIdentifier: t.product_identifier,
    unsubscribeDetectedAt: null,
    billingIssueDetectedAt: null,
    isSandbox: n.is_sandbox,
    periodType: "normal"
  };
}
function wc(e, t, n) {
  const r = {}, i = {};
  for (const o in e) {
    const a = bc(
      o,
      e[o],
      t,
      n
    );
    r[o] = a, a.isActive && (i[o] = a);
  }
  return {
    all: r,
    active: i
  };
}
function zt(e) {
  return e == null ? null : new Date(e);
}
function Ec(e) {
  const t = Pc(e), n = e.subscriber, r = Tc(
    n.non_subscriptions
  );
  return {
    entitlements: wc(
      n.entitlements,
      n.subscriptions,
      r
    ),
    allExpirationDatesByProduct: t,
    allPurchaseDatesByProduct: Ic(
      e,
      r
    ),
    activeSubscriptions: kc(t),
    managementURL: n.management_url,
    requestDate: new Date(e.request_date),
    firstSeenDate: new Date(n.first_seen),
    originalPurchaseDate: zt(
      n.original_purchase_date
    ),
    originalAppUserId: e.subscriber.original_app_user_id
  };
}
function xc(e, t) {
  if (t == null) return !1;
  const n = t.store == "promotional", r = e.expires_date == null, i = t.unsubscribe_detected_at != null, o = t.billing_issues_detected_at != null;
  return !(n || r || i || o);
}
function Ic(e, t) {
  const n = {};
  for (const r in e.subscriber.subscriptions) {
    const i = e.subscriber.subscriptions[r];
    n[r] = new Date(
      i.purchase_date
    );
  }
  for (const r in t) {
    const i = t[r].purchase_date;
    i == null ? n[r] = null : n[r] = new Date(i);
  }
  return n;
}
function Pc(e) {
  const t = {};
  for (const n in e.subscriber.subscriptions) {
    const r = e.subscriber.subscriptions[n];
    r.expires_date == null ? t[n] = null : t[n] = new Date(
      r.expires_date
    );
  }
  return t;
}
function kc(e) {
  const t = /* @__PURE__ */ new Set(), n = /* @__PURE__ */ new Date();
  for (const r in e) {
    const i = e[r];
    (i == null || i > n) && t.add(r);
  }
  return t;
}
function Tc(e) {
  const t = {};
  for (const n in e) {
    if (e[n].length === 0) continue;
    const r = e[n].length;
    t[n] = e[n][r - 1];
  }
  return t;
}
const Ac = "0.13.0", Oc = "https://api.revenuecat.com";
var Zt;
(function(e) {
  e[e.CONTINUE = 100] = "CONTINUE", e[e.SWITCHING_PROTOCOLS = 101] = "SWITCHING_PROTOCOLS", e[e.PROCESSING = 102] = "PROCESSING", e[e.EARLY_HINTS = 103] = "EARLY_HINTS", e[e.OK = 200] = "OK", e[e.CREATED = 201] = "CREATED", e[e.ACCEPTED = 202] = "ACCEPTED", e[e.NON_AUTHORITATIVE_INFORMATION = 203] = "NON_AUTHORITATIVE_INFORMATION", e[e.NO_CONTENT = 204] = "NO_CONTENT", e[e.RESET_CONTENT = 205] = "RESET_CONTENT", e[e.PARTIAL_CONTENT = 206] = "PARTIAL_CONTENT", e[e.MULTI_STATUS = 207] = "MULTI_STATUS", e[e.MULTIPLE_CHOICES = 300] = "MULTIPLE_CHOICES", e[e.MOVED_PERMANENTLY = 301] = "MOVED_PERMANENTLY", e[e.MOVED_TEMPORARILY = 302] = "MOVED_TEMPORARILY", e[e.SEE_OTHER = 303] = "SEE_OTHER", e[e.NOT_MODIFIED = 304] = "NOT_MODIFIED", e[e.USE_PROXY = 305] = "USE_PROXY", e[e.TEMPORARY_REDIRECT = 307] = "TEMPORARY_REDIRECT", e[e.PERMANENT_REDIRECT = 308] = "PERMANENT_REDIRECT", e[e.BAD_REQUEST = 400] = "BAD_REQUEST", e[e.UNAUTHORIZED = 401] = "UNAUTHORIZED", e[e.PAYMENT_REQUIRED = 402] = "PAYMENT_REQUIRED", e[e.FORBIDDEN = 403] = "FORBIDDEN", e[e.NOT_FOUND = 404] = "NOT_FOUND", e[e.METHOD_NOT_ALLOWED = 405] = "METHOD_NOT_ALLOWED", e[e.NOT_ACCEPTABLE = 406] = "NOT_ACCEPTABLE", e[e.PROXY_AUTHENTICATION_REQUIRED = 407] = "PROXY_AUTHENTICATION_REQUIRED", e[e.REQUEST_TIMEOUT = 408] = "REQUEST_TIMEOUT", e[e.CONFLICT = 409] = "CONFLICT", e[e.GONE = 410] = "GONE", e[e.LENGTH_REQUIRED = 411] = "LENGTH_REQUIRED", e[e.PRECONDITION_FAILED = 412] = "PRECONDITION_FAILED", e[e.REQUEST_TOO_LONG = 413] = "REQUEST_TOO_LONG", e[e.REQUEST_URI_TOO_LONG = 414] = "REQUEST_URI_TOO_LONG", e[e.UNSUPPORTED_MEDIA_TYPE = 415] = "UNSUPPORTED_MEDIA_TYPE", e[e.REQUESTED_RANGE_NOT_SATISFIABLE = 416] = "REQUESTED_RANGE_NOT_SATISFIABLE", e[e.EXPECTATION_FAILED = 417] = "EXPECTATION_FAILED", e[e.IM_A_TEAPOT = 418] = "IM_A_TEAPOT", e[e.INSUFFICIENT_SPACE_ON_RESOURCE = 419] = "INSUFFICIENT_SPACE_ON_RESOURCE", e[e.METHOD_FAILURE = 420] = "METHOD_FAILURE", e[e.MISDIRECTED_REQUEST = 421] = "MISDIRECTED_REQUEST", e[e.UNPROCESSABLE_ENTITY = 422] = "UNPROCESSABLE_ENTITY", e[e.LOCKED = 423] = "LOCKED", e[e.FAILED_DEPENDENCY = 424] = "FAILED_DEPENDENCY", e[e.UPGRADE_REQUIRED = 426] = "UPGRADE_REQUIRED", e[e.PRECONDITION_REQUIRED = 428] = "PRECONDITION_REQUIRED", e[e.TOO_MANY_REQUESTS = 429] = "TOO_MANY_REQUESTS", e[e.REQUEST_HEADER_FIELDS_TOO_LARGE = 431] = "REQUEST_HEADER_FIELDS_TOO_LARGE", e[e.UNAVAILABLE_FOR_LEGAL_REASONS = 451] = "UNAVAILABLE_FOR_LEGAL_REASONS", e[e.INTERNAL_SERVER_ERROR = 500] = "INTERNAL_SERVER_ERROR", e[e.NOT_IMPLEMENTED = 501] = "NOT_IMPLEMENTED", e[e.BAD_GATEWAY = 502] = "BAD_GATEWAY", e[e.SERVICE_UNAVAILABLE = 503] = "SERVICE_UNAVAILABLE", e[e.GATEWAY_TIMEOUT = 504] = "GATEWAY_TIMEOUT", e[e.HTTP_VERSION_NOT_SUPPORTED = 505] = "HTTP_VERSION_NOT_SUPPORTED", e[e.INSUFFICIENT_STORAGE = 507] = "INSUFFICIENT_STORAGE", e[e.NETWORK_AUTHENTICATION_REQUIRED = 511] = "NETWORK_AUTHENTICATION_REQUIRED";
})(Zt || (Zt = {}));
function Pn(e) {
  return e ? e.startsWith("rcb_sb_") : !1;
}
async function Je(e, t) {
  const { apiKey: n, body: r, headers: i, httpConfig: o } = t, s = `${(o == null ? void 0 : o.proxyURL) ?? Oc}${e.urlPath()}`;
  try {
    const c = await fetch(s, {
      method: e.method,
      headers: Lc(n, i, o == null ? void 0 : o.additionalHeaders),
      body: Sc(r)
    });
    return await Nc(c, e), await c.json();
  } catch (c) {
    throw c instanceof TypeError ? new Z(
      ne.NetworkError,
      xt.getPublicMessage(ne.NetworkError),
      c.message
    ) : c;
  }
}
async function Nc(e, t) {
  const n = e.status;
  if (n >= Zt.INTERNAL_SERVER_ERROR)
    vn(t, n, await e.text());
  else if (n >= Zt.BAD_REQUEST) {
    const r = await e.json(), i = r ? JSON.stringify(r) : null, o = r == null ? void 0 : r.code, a = r == null ? void 0 : r.message;
    if (o != null) {
      const s = xt.convertCodeToBackendErrorCode(o);
      if (s == null)
        vn(
          t,
          n,
          i,
          o
        );
      else
        throw Z.getForBackendError(
          s,
          a
        );
    } else
      vn(t, n, i);
  }
}
function vn(e, t, n, r) {
  throw new Z(
    ne.UnknownBackendError,
    "Unknown backend error.",
    `Request: ${e.name}. Status code: ${t}. Body: ${n}.`,
    { backendErrorCode: r }
  );
}
function Sc(e) {
  return e == null ? null : JSON.stringify(e);
}
const Bi = "Authorization", qi = "Content-Type", ji = "Accept", Fi = "X-Platform", Hi = "X-Version", Vi = "X-Is-Sandbox", Rc = /* @__PURE__ */ new Set([
  Bi,
  qi,
  ji,
  Fi,
  Hi,
  Vi
]);
function Lc(e, t, n) {
  let r = {
    [Bi]: `Bearer ${e}`,
    [qi]: "application/json",
    [ji]: "application/json",
    [Fi]: "web",
    [Hi]: Ac,
    [Vi]: `${Pn(e)}`
  };
  return t && (r = { ...r, ...t }), n && (r = { ...n, ...r }), r;
}
const Gi = "/v1/subscribers", ln = "/rcbilling/v1";
class Uc {
  constructor(t) {
    R(this, "appUserId");
    R(this, "method", "GET");
    R(this, "name", "getOfferings");
    this.appUserId = t;
  }
  urlPath() {
    const t = encodeURIComponent(this.appUserId);
    return `${Gi}/${t}/offerings`;
  }
}
class Dc {
  constructor() {
    R(this, "method", "POST");
    R(this, "name", "purchase");
  }
  urlPath() {
    return `${ln}/purchase`;
  }
}
class Mc {
  constructor(t, n, r) {
    R(this, "method", "GET");
    R(this, "name", "getProducts");
    R(this, "appUserId");
    R(this, "productIds");
    R(this, "currency");
    this.appUserId = t, this.productIds = n, this.currency = r;
  }
  urlPath() {
    const t = encodeURIComponent(this.appUserId), n = this.productIds.map(encodeURIComponent).join("&id="), r = this.currency ? `&currency=${this.currency}` : "";
    return `${ln}/subscribers/${t}/products?id=${n}${r}`;
  }
}
class Cc {
  constructor(t) {
    R(this, "method", "GET");
    R(this, "name", "getCustomerInfo");
    R(this, "appUserId");
    this.appUserId = t;
  }
  urlPath() {
    const t = encodeURIComponent(this.appUserId);
    return `${Gi}/${t}`;
  }
}
class Bc {
  constructor() {
    R(this, "method", "GET");
    R(this, "name", "getBrandingInfo");
  }
  urlPath() {
    return `${ln}/branding`;
  }
}
class qc {
  constructor(t) {
    R(this, "method", "GET");
    R(this, "name", "getCheckoutStatus");
    R(this, "operationSessionId");
    this.operationSessionId = t;
  }
  urlPath() {
    return `${ln}/checkout/${this.operationSessionId}`;
  }
}
const kn = {};
class jc {
  constructor(t, n = kn) {
    R(this, "API_KEY");
    R(this, "httpConfig");
    this.API_KEY = t, this.httpConfig = n;
  }
  async getOfferings(t) {
    return await Je(
      new Uc(t),
      {
        apiKey: this.API_KEY,
        httpConfig: this.httpConfig
      }
    );
  }
  async getCustomerInfo(t) {
    return await Je(
      new Cc(t),
      {
        apiKey: this.API_KEY,
        httpConfig: this.httpConfig
      }
    );
  }
  async getProducts(t, n, r) {
    return await Je(
      new Mc(t, n, r),
      {
        apiKey: this.API_KEY,
        httpConfig: this.httpConfig
      }
    );
  }
  async getBrandingInfo() {
    return await Je(
      new Bc(),
      {
        apiKey: this.API_KEY,
        httpConfig: this.httpConfig
      }
    );
  }
  async postPurchase(t, n, r, i, o) {
    const a = {
      app_user_id: t,
      product_id: n,
      email: r,
      price_id: o.priceId,
      presented_offering_identifier: i.offeringIdentifier
    };
    return o.id !== "base_option" && (a.offer_id = o.id), i.targetingContext && (a.applied_targeting_rule = {
      rule_id: i.targetingContext.ruleId,
      revision: i.targetingContext.revision
    }), i.placementIdentifier && (a.presented_placement_identifier = i.placementIdentifier), await Je(
      new Dc(),
      {
        apiKey: this.API_KEY,
        body: a,
        httpConfig: this.httpConfig
      }
    );
  }
  async getCheckoutStatus(t) {
    return await Je(
      new qc(t),
      {
        apiKey: this.API_KEY,
        httpConfig: this.httpConfig
      }
    );
  }
}
function Fc(e) {
  if (!/^rcb_[a-zA-Z0-9_.-]+$/.test(e))
    throw new Z(
      ne.InvalidCredentialsError,
      "Invalid API key. Use your RevenueCat Billing API key."
    );
}
function Hc(e) {
  if ((/* @__PURE__ */ new Set([
    "no_user",
    "null",
    "none",
    "nil",
    "(null)",
    "NaN",
    "\\x00",
    "",
    "unidentified",
    "undefined",
    "unknown"
  ])).has(e) || e.includes("/"))
    throw new Z(
      ne.InvalidAppUserIdError,
      'Provided user id: "' + e + '" is not valid. See https://www.revenuecat.com/docs/customers/user-ids#tips-for-setting-app-user-ids for more information.'
    );
}
function Vc(e) {
  if (e != null && e.endsWith("/"))
    throw new Z(
      ne.ConfigurationError,
      "Invalid proxy URL. The proxy URL should not end with a trailing slash."
    );
}
function Gc(e) {
  if (e) {
    for (const t in e)
      if (Rc.has(t))
        throw new Z(
          ne.ConfigurationError,
          "Invalid additional headers. Some headers are reserved for internal use."
        );
  }
}
var zi = /* @__PURE__ */ ((e) => (e.Current = "current", e))(zi || {});
function zc(e, t) {
  async function n(...r) {
    return await this.preload(), await e.call(this, ...r);
  }
  return n;
}
const Yi = (e, t) => ({
  ...e,
  rcBillingProduct: {
    ...e.rcBillingProduct,
    presentedOfferingContext: {
      ...e.rcBillingProduct.presentedOfferingContext,
      placementIdentifier: t
    }
  }
}), Ye = (e, t) => e == null ? null : Yi(e, t), Yc = (e, t, n) => {
  const r = e.offering_ids_by_placement ?? null;
  if (r == null)
    return null;
  const i = e.fallback_offering_id ?? null;
  let o;
  if (n in r) {
    const s = r[n] ?? null;
    if (s == null)
      return null;
    o = t[s], o == null && (o = t[i]);
  } else
    o = t[i];
  if (o == null)
    return null;
  const a = Object.fromEntries(
    Object.entries(o.packagesById).map(([s, c]) => [
      s,
      Yi(c, n)
    ])
  );
  return {
    ...o,
    packagesById: a,
    availablePackages: Object.values(a),
    weekly: Ye(o.weekly, n),
    monthly: Ye(
      o.monthly,
      n
    ),
    twoMonth: Ye(
      o.twoMonth,
      n
    ),
    threeMonth: Ye(
      o.threeMonth,
      n
    ),
    sixMonth: Ye(
      o.sixMonth,
      n
    ),
    annual: Ye(o.annual, n),
    lifetime: Ye(
      o.lifetime,
      n
    )
  };
};
function $c(e, t) {
  const n = {};
  t.product_details.forEach((o) => {
    n[o.identifier] = o;
  });
  const r = {};
  e.offerings.forEach((o) => {
    const a = o.identifier === e.current_offering_id, s = Ya(
      a,
      o,
      n,
      e.targeting
    );
    s != null && (r[o.identifier] = s);
  });
  const i = e.current_offering_id ? r[e.current_offering_id] ?? null : null;
  return Object.keys(r).length == 0 && F.debugLog(
    "Empty offerings. Please make sure you've configured offerings correctly in the RevenueCat dashboard and that the products are properly configured."
  ), {
    all: r,
    current: i
  };
}
var Tn;
(function(e) {
  e.body_s = "p", e.body_m = "p", e.body_l = "p", e.body_xl = "p", e.heading_xs = "h6", e.heading_s = "h5", e.heading_m = "h4", e.heading_l = "h3", e.heading_xl = "h2", e.heading_xxl = "h1";
})(Tn || (Tn = {}));
var An;
(function(e) {
  e[e.extra_light = 100] = "extra_light", e[e.thin = 200] = "thin", e[e.light = 300] = "light", e[e.regular = 400] = "regular", e[e.medium = 500] = "medium", e[e.semi_bold = 600] = "semi_bold", e[e.bold = 700] = "bold", e[e.extra_bold = 800] = "extra_bold", e[e.black = 900] = "black";
})(An || (An = {}));
var On;
(function(e) {
  e.body_s = "13px", e.body_m = "14px", e.body_l = "15px", e.body_xl = "16px", e.heading_xs = "14px", e.heading_s = "16px", e.heading_m = "18px", e.heading_l = "20px", e.heading_xl = "26px", e.heading_xxl = "32px";
})(On || (On = {}));
var Nn;
(function(e) {
  e.vertical = "column", e.horizontal = "row";
})(Nn || (Nn = {}));
var Sn;
(function(e) {
  e.top = "flex-start", e.leading = "flex-start", e.center = "center", e.bottom = "flex-end", e.trailing = "flex-end";
})(Sn || (Sn = {}));
var Rn;
(function(e) {
  e.space_between = "space-between", e.space_around = "space-around", e.space_evenly = "space-evenly", e.start = "flex-start", e.center = "center", e.end = "flex-end";
})(Rn || (Rn = {}));
const et = [];
function Kc(e, t) {
  return {
    subscribe: Qc(e, t).subscribe
  };
}
function Qc(e, t = Ce) {
  let n = null;
  const r = /* @__PURE__ */ new Set();
  function i(s) {
    if (zr(e, s) && (e = s, n)) {
      const c = !et.length;
      for (const l of r)
        l[1](), et.push(l, e);
      if (c) {
        for (let l = 0; l < et.length; l += 2)
          et[l][0](et[l + 1]);
        et.length = 0;
      }
    }
  }
  function o(s) {
    i(s(
      /** @type {T} */
      e
    ));
  }
  function a(s, c = Ce) {
    const l = [s, c];
    return r.add(l), r.size === 1 && (n = t(i, o) || Ce), s(
      /** @type {T} */
      e
    ), () => {
      r.delete(l), r.size === 0 && n && (n(), n = null);
    };
  }
  return { set: i, update: o, subscribe: a };
}
const Wc = Kc(!1, (e) => {
  if (typeof window < "u") {
    const t = window.matchMedia("(prefers-color-scheme: dark)");
    e(t.matches);
    const n = (r) => e(r.matches);
    return t.addEventListener("change", n), () => t.removeEventListener("change", n);
  }
  return () => {
  };
}), Zc = (e, t) => {
  var n;
  (n = t.onclick) == null || n.call(t);
};
var Xc = /* @__PURE__ */ I("<button><!></button>");
const Jc = {
  hash: "svelte-8axjuk",
  code: "button.svelte-8axjuk {border:none;cursor:pointer;background-color:transparent;padding:0;width:100%;display:flex;align-items:center;justify-content:center;}"
};
function $i(e, t) {
  Y(t, !0), D(e, Jc);
  var n = Xc();
  n.__click = [Zc, t];
  var r = k(n);
  pa(r, () => t.children ?? Ce), B(() => {
    Yn(n, `${t.class ?? ""} svelte-8axjuk`), n.disabled = !t.onclick, G(n, "style", t.style);
  }), _(e, n), $();
}
zn(["click"]);
function Mr(e, t) {
  return !e || !t ? "" : {
    [`--${t}-block-start`]: `${e.top ?? 0}px`,
    [`--${t}-inline-end`]: `${e.trailing ?? 0}px`,
    [`--${t}-block-end`]: `${e.bottom ?? 0}px`,
    [`--${t}-inline-start`]: `${e.leading ?? 0}px`
  };
}
function eu(e) {
  return Tn[e];
}
function Yt({ colorMap: e, colorMode: t = "light", fallback: n = "FFFFFF" }) {
  var r;
  return ((r = e == null ? void 0 : e[t]) == null ? void 0 : r.value) || n;
}
function tu(e, t = "light") {
  const n = Yt({ colorMap: e.color, colorMode: t });
  return `${e.width}px solid ${n}`;
}
function Ki(e) {
  return `${e.top_leading}px ${e.top_trailing}px ${e.bottom_trailing}px ${e.bottom_leading}px`;
}
function Qi({ backgroundColor: e, border: t, margin: n, padding: r, textColor: i, colorMode: o = "light", shape: a, shadow: s }) {
  const c = {
    "--margin-block-start": "0px",
    "--margin-inline-end": "0px",
    "--margin-block-end": "0px",
    "--margin-inline-start": "0px",
    "--padding-block-start": "0px",
    "--padding-inline-end": "0px",
    "--padding-block-end": "0px",
    "--padding-inline-start": "0px",
    "--background-color": "transparent",
    "--text-color": "#000000",
    "--border": "none",
    "--border-radius": "0px",
    "--shadow": "none"
  };
  return r && Object.assign(c, Mr(r, "padding")), n && Object.assign(c, Mr(n, "margin")), e && (c["--background-color"] = Yt({
    colorMap: e,
    colorMode: o,
    fallback: "transparent"
  })), i && (c["--text-color"] = Yt({
    colorMap: i,
    colorMode: o,
    fallback: "#000000"
  })), t && (c["--border"] = tu(t, o)), (a == null ? void 0 : a.type) === "rectangle" && a.corners && (c["--border-radius"] = Ki(a.corners)), (a == null ? void 0 : a.type) === "pill" && (c["--border-radius"] = "9999px;"), s && (c["--shadow"] = `${s.x}px ${s.y}px ${s.radius}px 
      ${Yt({ colorMap: s.color, colorMode: o, fallback: "#000000" })};`), c;
}
function er(e) {
  function t(o) {
    return o.type === "fixed" ? `${o.value}px` : o.type === "fit" ? "fit-content" : o.type === "fill" ? navigator.userAgent.match(/firefox|fxios/i) ? "-moz-available" : "-webkit-fill-available" : "unset";
  }
  const n = t(e.width), r = t(e.height), i = e.width.type === "fill" || e.height.type === "fill";
  return {
    "--width": n,
    "--height": r,
    "--flex": i ? "unset" : "0 1 auto"
  };
}
function nu(e, t) {
  return t ? {
    "--background": `linear-gradient(${t.map((n) => {
      var r;
      return (r = n[e]) == null ? void 0 : r.value;
    }).join(", ")})`
  } : { "--background": "unset" };
}
const ru = (e) => {
  let t = {
    "--corner-radius": "0px",
    "--clip-path": "none"
  };
  return (e == null ? void 0 : e.type) === "circle" && (t["--corner-radius"] = "50%"), (e == null ? void 0 : e.type) === "rectangle" && e.corners && (t["--corner-radius"] = Ki(e.corners)), (e == null ? void 0 : e.type) === "convex" && (t["--corner-radius"] = "0 0 50% 50%;"), (e == null ? void 0 : e.type) === "concave" && (t["--clip-path"] = "polygon(0 0, 100% 0, 100% 100%, 50% 110%, 0 100%);"), t;
};
function iu(e) {
  const t = {
    "--direction": "row",
    "--alignment": "unset",
    "--distribution": "unset"
  };
  return e.type === "zlayer" || (t["--direction"] = Nn[e.type], t["--alignment"] = Sn[e.alignment], e.distribution && (t["--distribution"] = Rn[e.distribution])), t;
}
function ou(e) {
  const { font_size: t, horizontal_alignment: n, font_weight: r, font_name: i } = e, o = {
    "--text-align": "initial",
    "--font-weight": 400,
    "--font-size": "initial",
    "--font-family": "sans-serif"
  };
  return o["--text-align"] = n, o["--font-weight"] = An[r], o["--font-size"] = On[t], o["--font-family"] = i || "sans-serif", o;
}
function Xt(e) {
  return Object.entries(e).map(([t, n]) => `${t}: ${n};`).join("; ");
}
function Wi(e, t, n, r) {
  const i = (r[n] || {})[e];
  return (r[t] || {})[e] ? r[t][e] : i;
}
function au(e) {
  const t = (r) => {
    if (r.type === "package" && r.is_selected_by_default)
      return r;
    if (r.components)
      for (let i of r.components) {
        const o = t(i);
        if (o)
          return o;
      }
    if (r.stack !== void 0) {
      const i = t(r.stack);
      if (i)
        return i;
    }
  }, n = t(e.components_config.base.stack);
  if (n !== void 0)
    return n.package_id;
}
const su = (e) => {
  const { background_color: t, margin: n, padding: r, spacing: i, size: o, border: a, shape: s, colorMode: c, shadow: l, dimension: d } = e, u = Qi({
    backgroundColor: t,
    margin: n,
    padding: r,
    border: a,
    colorMode: c,
    shape: s,
    shadow: l
  }), f = iu(d);
  return {
    ...er(o),
    ...f,
    ...u,
    "--spacing": `${i || 0}px`
  };
};
var lu = /* @__PURE__ */ I('<div class="stack svelte-1igbv3z"></div>');
const cu = {
  hash: "svelte-1igbv3z",
  code: ".stack.svelte-1igbv3z {box-sizing:border-box;display:flex;flex-direction:var(--direction, unset);width:var(--width, unset);height:var(--height, unset);gap:var(--spacing, 0);background-color:var(--background-color, unset);border:var(--border, none);border-radius:var(--border-radius, 0);box-shadow:var(--shadow, none);justify-content:var(--distribution, flex-start);align-items:var(--alignment, flex-start);padding-block-start:var(--padding-block-start, 0);padding-inline-end:var(--padding-inline-end, 0);padding-block-end:var(--padding-block-end, 0);padding-inline-start:var(--padding-inline-start, 0);margin-block-start:var(--margin-block-start, 0);margin-inline-end:var(--margin-inline-end, 0);margin-block-end:var(--margin-block-end, 0);margin-inline-start:var(--margin-inline-start, 0);flex:var(--flex, 0 1 auto);align-self:center;}"
};
function qt(e, t) {
  Y(t, !0), D(e, cu);
  const n = /* @__PURE__ */ sn(t, [
    "$$slots",
    "$$events",
    "$$legacy",
    "id",
    "components",
    "labels",
    "onAction",
    "purchaseState"
  ]), r = /* @__PURE__ */ K(() => Xt(su({
    id: t.id,
    components: t.components,
    labels: t.labels,
    onAction: t.onAction,
    purchaseState: t.purchaseState,
    ...n
  })));
  var i = lu();
  hi(i, 21, () => t.components, gi, (o, a) => {
    tr(o, {
      get nodeData() {
        return g(a);
      },
      get labels() {
        return t.labels;
      },
      get onAction() {
        return t.onAction;
      },
      get purchaseState() {
        return t.purchaseState;
      }
    });
  }), B(() => {
    G(i, "id", t.id), G(i, "style", g(r));
  }), _(e, i), $();
}
function uu(e, t) {
  Y(t, !0), $i(e, {
    onclick: () => {
      t.onAction && t.onAction(t.action);
    },
    children: (r, i) => {
      qt(r, Ct(() => t.stack, {
        get labels() {
          return t.labels;
        },
        get purchaseState() {
          return t.purchaseState;
        }
      }));
    },
    $$slots: { default: !0 }
  }), $();
}
function tr(e, t) {
  Y(t, !0);
  const n = Ta(), r = () => ka(Wc, "$prefersDarkMode", n), i = {
    text: pu,
    stack: qt,
    image: yu,
    button: uu,
    purchase_button: Eu,
    package: ku,
    footer: Au
  }, o = /* @__PURE__ */ K(() => i[t.nodeData.type]);
  var a = te(), s = U(a);
  _a(s, () => g(o), (c, l) => {
    l(c, Ct(() => t.nodeData || {}, {
      get labels() {
        return t.labels;
      },
      get prefersDarkMode() {
        return r();
      },
      get onAction() {
        return t.onAction;
      },
      get purchaseState() {
        return t.purchaseState;
      },
      children: (d, u) => {
        var f = te(), v = U(f);
        hi(v, 17, () => t.nodeData.components, gi, (p, b) => {
          tr(p, {
            get nodeData() {
              return g(b);
            },
            get labels() {
              return t.labels;
            },
            get onAction() {
              return t.onAction;
            },
            get purchaseState() {
              return t.purchaseState;
            }
          });
        }), _(d, f);
      },
      $$slots: { default: !0 }
    }));
  }), _(e, a), $();
}
var fu = /* @__PURE__ */ I('<div class="svelte-1f98y3q"><!></div>');
const du = {
  hash: "svelte-1f98y3q",
  code: `
  /* 1. Use a more-intuitive box-sizing model */.svelte-1f98y3q,
  .svelte-1f98y3q::before,
  .svelte-1f98y3q::after {box-sizing:border-box;}

  /* 2. Remove default margin */.svelte-1f98y3q {margin:0;}

  /* 5. Improve media defaults */

  /* 6. Inherit fonts for form controls */

  /* 7. Avoid text overflows */

  /* 8. Improve line wrapping */

  /*
          9. Create a root stacking context
        */`
};
function vu(e, t) {
  Y(t, !0), D(e, du);
  let n = $e({
    selectedPackageId: au(t.paywallData),
    locale: t.selectedLocale || t.paywallData.default_locale,
    defaultLocale: t.paywallData.default_locale
  });
  const r = (u, f) => {
    var v;
    (v = t.onRestorePurchasesClicked) == null || v.call(t);
  }, i = (u, f) => {
    var v;
    (v = t.onBackClicked) == null || v.call(t);
  }, o = (u, f) => {
    var p, b;
    if (u.destination === "customer_center" && ((p = t.onVisitCustomerCenterClicked) == null || p.call(t)), !u.url)
      return;
    const v = Wi(u.url.url_lid, n.locale, n.defaultLocale, t.paywallData.components_localizations);
    v && ((b = t.onNavigateToUrlClicked) == null || b.call(t, v));
  }, a = (u, f) => {
    n.selectedPackageId = f == null ? void 0 : f.packageId;
  }, s = (u, f) => {
    var v;
    n.selectedPackageId !== void 0 && ((v = t.onPurchaseClicked) == null || v.call(t, n.selectedPackageId));
  }, c = (u, f) => {
    switch (u.type) {
      case "select_package":
        a(u, f);
        break;
      case "navigate_to":
        o(u);
        break;
      case "navigate_back":
        i();
        break;
      case "restore_purchases":
        r();
        break;
      case "purchase":
        s();
        break;
    }
  };
  var l = fu(), d = k(l);
  tr(d, {
    onAction: c,
    get nodeData() {
      return t.paywallData.components_config.base.stack;
    },
    get labels() {
      return t.paywallData.components_localizations;
    },
    get purchaseState() {
      return n;
    }
  }), _(e, l), $();
}
const gu = (e) => {
  const { font_size: t = "body_m", color: n = {
    dark: { type: "hex", value: "#000000" },
    light: { type: "hex", value: "#000000" }
  }, background_color: r = {
    dark: { type: "hex", value: "#FFFFFF" },
    light: { type: "hex", value: "#FFFFFF" }
  }, padding: i, margin: o, colorMode: a, size: s } = e, c = eu(t), l = ou(e), d = Qi({
    backgroundColor: r,
    textColor: n,
    margin: o,
    padding: i,
    colorMode: a
  }), u = er(s);
  return {
    textStyles: {
      ...l,
      ...d,
      ...u
    },
    tagToRender: c
  };
}, hu = {
  hash: "svelte-10xncm6",
  code: ".text-block.svelte-10xncm6 {padding-block-start:var(--padding-block-start, unset);padding-inline-end:var(--padding-inline-end, unset);padding-block-end:var(--padding-block-end, unset);padding-inline-start:var(--padding-inline-start, unset);margin-block-start:var(--margin-block-start, unset);margin-inline-end:var(--margin-inline-end, unset);margin-block-end:var(--margin-block-end, unset);margin-inline-start:var(--margin-inline-start, unset);color:var(--text-color, #ffffff);background-color:var(--background-color, #ffffff);font-size:var(--font-size, 1rem);font-weight:var(--font-weight, normal);text-align:var(--text-align, center);height:var(--height, unset);width:var(--width, unset);font-family:var(--font-family, sans-serif);}"
};
function pu(e, t) {
  Y(t, !0), D(e, hu);
  const n = /* @__PURE__ */ sn(t, [
    "$$slots",
    "$$events",
    "$$legacy",
    "id",
    "labels",
    "text_lid",
    "purchaseState"
  ]), r = /* @__PURE__ */ K(() => gu({
    id: t.id,
    labels: t.labels,
    text_lid: t.text_lid,
    purchaseState: t.purchaseState,
    ...n
  })), i = /* @__PURE__ */ K(() => g(r).tagToRender), o = /* @__PURE__ */ K(() => g(r).textStyles), a = /* @__PURE__ */ K(() => Xt(g(o))), s = /* @__PURE__ */ K(() => Wi(t.text_lid, t.purchaseState.locale, t.purchaseState.defaultLocale, t.labels));
  var c = te(), l = U(c);
  pi(l, () => g(i), !1, (d, u) => {
    let f;
    B(() => f = _i(
      d,
      f,
      {
        id: t.id,
        style: g(a),
        class: "text-block"
      },
      "svelte-10xncm6",
      d.namespaceURI === Dn,
      d.nodeName.includes("-")
    ));
    var v = He();
    B(() => re(v, g(s))), _(u, v);
  }), _(e, c), $();
}
const _u = (e) => {
  const {
    gradient_colors: t,
    colorMode: n,
    mask_shape: r,
    size: i,
    max_height: o,
    // TODO: implement this. still waiting on spec
    override_source_lid: a
    // TODO: Implement this once structure is defined
  } = e, s = nu(n, t), c = er(i), l = ru(r);
  return { imageStyles: {
    ...c,
    ...l
  }, gradientStyles: s };
};
var bu = /* @__PURE__ */ I('<div class="image-container svelte-1p87anq"><img class="image svelte-1p87anq" alt=""> <span class="image-overlay svelte-1p87anq"></span></div>');
const mu = {
  hash: "svelte-1p87anq",
  code: ".image-container.svelte-1p87anq {position:relative;height:var(--height, unset);width:var(--width, unset);overflow:hidden;border-radius:var(--corner-radius, unset);clip-path:var(--clip-path, unset);display:flex;flex:1 0 auto;}.image.svelte-1p87anq {width:100%;height:100%;object-fit:cover;}.image-overlay.svelte-1p87anq {position:absolute;inset:0;background:var(--background, unset);}"
};
function yu(e, t) {
  Y(t, !0), D(e, mu);
  const n = /* @__PURE__ */ sn(t, [
    "$$slots",
    "$$events",
    "$$legacy",
    "id",
    "colorMode",
    "source"
  ]), r = /* @__PURE__ */ K(() => _u({
    id: t.id,
    colorMode: t.colorMode,
    source: t.source,
    ...n
  })), i = /* @__PURE__ */ K(() => g(r).gradientStyles), o = /* @__PURE__ */ K(() => g(r).imageStyles), a = /* @__PURE__ */ K(() => Xt(g(i))), s = /* @__PURE__ */ K(() => Xt(g(o)));
  var c = bu(), l = k(c), d = N(l, 2);
  B(() => {
    var u;
    G(c, "id", `image-container-${t.id}`), G(c, "style", g(s)), G(l, "src", (u = t.source[t.colorMode || "light"]) == null ? void 0 : u.original), G(l, "id", t.id), G(d, "style", g(a));
  }), _(e, c), $();
}
const wu = { hash: "svelte-1rjn6eg", code: "" };
function Eu(e, t) {
  Y(t, !0), D(e, wu), $i(e, {
    onclick: () => {
      t.onAction && t.onAction({ type: "purchase" });
    },
    class: "purchase-button",
    children: (r, i) => {
      qt(r, Ct(() => t.stack, {
        get labels() {
          return t.labels;
        },
        get purchaseState() {
          return t.purchaseState;
        }
      }));
    },
    $$slots: { default: !0 }
  }), $();
}
const xu = (e, t) => {
  e.preventDefault(), e.stopPropagation(), t.onAction && t.onAction({ type: "select_package" }, { packageId: t.package_id });
};
var Iu = /* @__PURE__ */ I('<div class="package svelte-j5j2a5"><!></div>');
const Pu = {
  hash: "svelte-j5j2a5",
  code: ".package.svelte-j5j2a5 {width:100%;display:flex;cursor:pointer;}"
};
function ku(e, t) {
  Y(t, !0), D(e, Pu);
  const n = /* @__PURE__ */ K(() => {
    var o;
    return ((o = t.purchaseState) == null ? void 0 : o.selectedPackageId) === t.package_id;
  });
  var r = Iu();
  r.__click = [xu, t];
  var i = k(r);
  qt(i, Ct(() => t.stack, {
    get labels() {
      return t.labels;
    },
    get onAction() {
      return t.onAction;
    },
    get purchaseState() {
      return t.purchaseState;
    }
  })), B(() => {
    G(r, "id", t.id), ma(r, "selected", g(n));
  }), _(e, r), $();
}
zn(["click"]);
var Tu = /* @__PURE__ */ I('<div class="footer"><!></div>');
function Au(e, t) {
  const n = /* @__PURE__ */ sn(t, ["$$slots", "$$events", "$$legacy"]), r = /* @__PURE__ */ K(() => n.stack), i = /* @__PURE__ */ K(() => n.labels), o = /* @__PURE__ */ K(() => n.id);
  var a = Tu(), s = k(a);
  qt(s, Ct(() => g(r), {
    get labels() {
      return g(i);
    }
  })), B(() => G(a, "id", g(o))), _(e, a);
}
const Ou = 1000001;
var Br, gn;
Br = [zc];
const le = class le {
  /** @internal */
  constructor(t, n, r = kn) {
    ur(gn, 5, this);
    /** @internal */
    R(this, "_API_KEY");
    /** @internal */
    R(this, "_appUserId");
    /** @internal */
    R(this, "_brandingInfo", null);
    /** @internal */
    R(this, "_loadingResourcesPromise", null);
    /** @internal */
    R(this, "backend");
    /** @internal */
    R(this, "purchaseOperationHelper");
    this._API_KEY = t, this._appUserId = n, Pn(t) && F.debugLog("Initializing Purchases SDK with sandbox API Key"), this.backend = new jc(this._API_KEY, r), this.purchaseOperationHelper = new us(this.backend);
  }
  /**
   * Set the log level. Logs of the given level and below will be printed
   * in the console.
   * Default is `LogLevel.Silent` so no logs will be printed in the console.
   * @param logLevel - LogLevel to set.
   */
  static setLogLevel(t) {
    F.setLogLevel(t);
  }
  /**
   * Get the singleton instance of Purchases. It's preferred to use the instance
   * obtained from the {@link Purchases.configure} method when possible.
   * @throws {@link UninitializedPurchasesError} if the instance has not been initialized yet.
   */
  static getSharedInstance() {
    if (le.isConfigured())
      return le.instance;
    throw new ls();
  }
  /**
   * Returns whether the Purchases SDK is configured or not.
   */
  static isConfigured() {
    return le.instance !== void 0;
  }
  /**
   * Configures the Purchases SDK. This should be called as soon as your app
   * has a unique user id for your user. You should only call this once, and
   * keep the returned instance around for use throughout your application.
   * @param apiKey - RevenueCat API Key. Can be obtained from the RevenueCat dashboard.
   * @param appUserId - Your unique id for identifying the user.
   * @param httpConfig - Advanced http configuration to customise the SDK usage {@link HttpConfig}.
   * @throws {@link PurchasesError} if the API key or user id are invalid.
   */
  static configure(t, n, r = kn) {
    return le.instance !== void 0 && F.warnLog(
      "Purchases is already initialized. You normally should only configure Purchases once. Creating and returning new instance."
    ), Fc(t), Hc(n), Vc(r.proxyURL), Gc(r.additionalHeaders), le.instance = new le(t, n, r), le.getSharedInstance();
  }
  /**
   * Loads and caches some optional data in the Purchases SDK.
   * Currently only fetching branding information. You can call this method
   * after configuring the SDK to speed up the first call to
   * {@link Purchases.purchase}.
   */
  async preload() {
    if (this.hasLoadedResources()) {
      F.verboseLog("Purchases resources are loaded. Skipping.");
      return;
    }
    if (this._loadingResourcesPromise !== null) {
      F.verboseLog("Purchases resources are loading. Waiting."), await this._loadingResourcesPromise;
      return;
    }
    return this._loadingResourcesPromise = this.backend.getBrandingInfo().then((t) => {
      this._brandingInfo = t;
    }).catch((t) => {
      let n = `${t}`;
      t instanceof Z && (n = `${t.message}. ${t.underlyingErrorMessage ? `Underlying error: ${t.underlyingErrorMessage}` : ""}`), F.errorLog(`Error fetching branding info: ${n}`);
    }).finally(() => {
      this._loadingResourcesPromise = null;
    }), this._loadingResourcesPromise;
  }
  hasLoadedResources() {
    return this._brandingInfo !== null;
  }
  /**
   * Renders an RC Paywall and allows the user to purchase from it using RCBilling.
   * @experimental
   * @internal
   * @param paywallParams - The parameters object to customise the paywall render. Check {@link RenderPaywallParams}
   * @returns Promise<PurchaseResult>
   */
  async renderPaywall(t) {
    console.warn(
      "This method is @experimental, Paywalls are not generally available but they will come soon!"
    );
    let r = t.htmlTarget ?? document.getElementById("rcb-ui-pw-root");
    if (r === null) {
      const u = document.createElement("div");
      u.id = "rcb-ui-pw-root", u.className = "rcb-ui-pw-root", u.style.zIndex = `${Ou}`, document.body.appendChild(u), r = u;
    }
    if (r === null)
      throw new Error(
        "Could not generate a mount point for the billing widget"
      );
    const i = r;
    i.innerHTML = "";
    const o = t.offering;
    if (!o.paywall_components)
      throw new Error("You cannot use paywalls yet, they are coming soon!");
    const a = (u) => {
      const f = o.availablePackages.find(
        (v) => v.identifier === u
      );
      if (f === void 0)
        throw new Error(`No package found for ${u}`);
      return this.purchase({
        rcPackage: f,
        htmlTarget: t.purchaseHtmlTarget,
        customerEmail: t.customerEmail
      });
    }, s = (u) => {
      var f;
      if (t.onNavigateToUrl) {
        t.onNavigateToUrl(u);
        return;
      }
      (f = window.open(u, "_blank")) == null || f.focus();
    }, c = () => {
    }, l = () => {
      if (t.onVisitCustomerCenter) {
        t.onVisitCustomerCenter();
        return;
      }
    }, d = t.selectedLocale || navigator.language;
    return new Promise((u, f) => {
      br(vu, {
        target: i,
        props: {
          paywallData: o.paywall_components,
          selectedLocale: d,
          onNavigateToClicked: s,
          onVisitCustomerCenterClicked: l,
          onBackClicked: () => {
            if (t.onBack) {
              t.onBack();
              return;
            }
            i.innerHTML = "", F.debugLog("Purchase cancelled by user"), f(new Z(ne.UserCancelledError));
          },
          onRestorePurchasesClicked: c,
          onPurchaseClicked: (v) => {
            a(v).then((p) => {
              u(p);
            }).catch((p) => f(p));
          }
        }
      });
    });
  }
  /**
   * Fetch the configured offerings for this user. You can configure these
   * in the RevenueCat dashboard.
   * @param params - The parameters object to customise the offerings fetch. Check {@link GetOfferingsParams}
   */
  async getOfferings(t) {
    Ul(t == null ? void 0 : t.currency);
    const n = this._appUserId, r = await this.backend.getOfferings(n), i = (t == null ? void 0 : t.offeringIdentifier) === zi.Current ? r.current_offering_id : t == null ? void 0 : t.offeringIdentifier;
    return i && (r.offerings = r.offerings.filter(
      (o) => o.identifier === i
    )), await this.getAllOfferings(r, n, t);
  }
  /**
   * Retrieves a specific offering by a placement identifier.
   * For more info see https://www.revenuecat.com/docs/tools/targeting
   * @param placementIdentifier - The placement identifier to retrieve the offering for.
   * @param params - The parameters object to customise the offerings fetch. Check {@link GetOfferingsParams}
   */
  async getCurrentOfferingForPlacement(t, n) {
    const r = this._appUserId, i = await this.backend.getOfferings(r), o = await this.getAllOfferings(
      i,
      r,
      n
    ), a = i.placements ?? null;
    return a == null ? null : Yc(
      a,
      o.all,
      t
    );
  }
  async getAllOfferings(t, n, r) {
    const i = t.offerings.flatMap((a) => a.packages).map((a) => a.platform_product_identifier), o = await this.backend.getProducts(
      n,
      i,
      r == null ? void 0 : r.currency
    );
    return this.logMissingProductIds(i, o.product_details), $c(t, o);
  }
  /**
   * Convenience method to check whether a user is entitled to a specific
   * entitlement. This will use {@link Purchases.getCustomerInfo} under the hood.
   * @param entitlementIdentifier - The entitlement identifier you want to check.
   * @returns Whether the user is entitled to the specified entitlement
   * @throws {@link PurchasesError} if there is an error while fetching the customer info.
   * @see {@link Purchases.getCustomerInfo}
   */
  async isEntitledTo(t) {
    const n = await this.getCustomerInfo();
    return t in n.entitlements.active;
  }
  /**
   * Method to perform a purchase for a given package. You can obtain the
   * package from {@link Purchases.getOfferings}. This method will present the purchase
   * form on your site, using the given HTML element as the mount point, if
   * provided, or as a modal if not.
   * @deprecated - please use .purchase
   * @param rcPackage - The package you want to purchase. Obtained from {@link Purchases.getOfferings}.
   * @param customerEmail - The email of the user. If undefined, RevenueCat will ask the customer for their email.
   * @param htmlTarget - The HTML element where the billing view should be added. If undefined, a new div will be created at the root of the page and appended to the body.
   * @returns a Promise for the customer info after the purchase is completed successfully.
   * @throws {@link PurchasesError} if there is an error while performing the purchase. If the {@link PurchasesError.errorCode} is {@link ErrorCode.UserCancelledError}, the user cancelled the purchase.
   */
  purchasePackage(t, n, r) {
    return this.purchase({
      rcPackage: t,
      customerEmail: n,
      htmlTarget: r
    });
  }
  purchase(t) {
    const { rcPackage: n, purchaseOption: r, htmlTarget: i, customerEmail: o } = t;
    let a = i ?? document.getElementById("rcb-ui-root");
    if (a === null) {
      const d = document.createElement("div");
      d.className = "rcb-ui-root", document.body.appendChild(d), a = d;
    }
    if (a === null)
      throw new Error(
        "Could not generate a mount point for the billing widget"
      );
    const s = a, c = !i, l = this._appUserId;
    return F.debugLog(
      `Presenting purchase form for package ${n.identifier}`
    ), new Promise((d, u) => {
      br(pc, {
        target: s,
        props: {
          appUserId: l,
          rcPackage: n,
          purchaseOption: r,
          customerEmail: o,
          onFinished: async (f) => {
            F.debugLog("Purchase finished"), s.innerHTML = "";
            const v = {
              customerInfo: await this._getCustomerInfoForUserId(l),
              redemptionInfo: f
            };
            d(v);
          },
          onClose: () => {
            s.innerHTML = "", F.debugLog("Purchase cancelled by user"), u(new Z(ne.UserCancelledError));
          },
          onError: (f) => {
            s.innerHTML = "", u(Z.getForPurchasesFlowError(f));
          },
          purchases: this,
          brandingInfo: this._brandingInfo,
          purchaseOperationHelper: this.purchaseOperationHelper,
          asModal: c
        }
      });
    });
  }
  /**
   * Gets latest available {@link CustomerInfo}.
   * @returns The latest {@link CustomerInfo}.
   * @throws {@link PurchasesError} if there is an error while fetching the customer info.
   */
  async getCustomerInfo() {
    return await this._getCustomerInfoForUserId(this._appUserId);
  }
  /**
   * Gets the current app user id.
   */
  getAppUserId() {
    return this._appUserId;
  }
  /**
   * Change the current app user id. Returns the customer info for the new
   * user id.
   * @param newAppUserId - The user id to change to.
   */
  async changeUser(t) {
    return this._appUserId = t, await this.getCustomerInfo();
  }
  /** @internal */
  logMissingProductIds(t, n) {
    const r = {};
    n.forEach(
      (o) => r[o.identifier] = o
    );
    const i = [];
    t.forEach((o) => {
      r[o] === void 0 && i.push(o);
    }), i.length > 0 && F.debugLog(
      `Could not find product data for product ids:
        ${i.join()}.
        Please check that your product configuration is correct.`
    );
  }
  /**
   * @returns Whether the SDK is using a sandbox API Key.
   */
  isSandbox() {
    return Pn(this._API_KEY);
  }
  /**
   * Closes the Purchases instance. You should never have to do this normally.
   */
  close() {
    le.instance === this ? le.instance = void 0 : F.warnLog(
      "Trying to close a Purchases instance that is not the current instance. Ignoring."
    );
  }
  /** @internal */
  async _getCustomerInfoForUserId(t) {
    const n = await this.backend.getCustomerInfo(t);
    return Ec(n);
  }
  /**
   * Generates an anonymous app user ID that follows RevenueCat's format.
   * This can be used when you don't have a user identifier system in place.
   * The generated ID will be in the format: $RCAnonymousID:<UUID without dashes>
   * Example: $RCAnonymousID:123e4567e89b12d3a456426614174000
   * @returns A new anonymous app user ID string
   * @public
   */
  static generateRevenueCatAnonymousAppUserId() {
    return `$RCAnonymousID:${crypto.randomUUID().replace(/-/g, "")}`;
  }
};
gn = lr(null), fr(gn, 1, "purchase", Br, le), fn(gn, le), /** @internal */
R(le, "instance");
let Cr = le;
export {
  ne as ErrorCode,
  de as LogLevel,
  zi as OfferingKeyword,
  qa as PackageType,
  Bt as ProductType,
  Cr as Purchases,
  Z as PurchasesError,
  ls as UninitializedPurchasesError
};
